<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Build a GenomicState object for Gencode TxDb objects — gencode_genomic_state • GenomicState</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Build a GenomicState object for Gencode TxDb objects — gencode_genomic_state" />
<meta property="og:description" content="Based on a TxDb object built by gencode_txdb() this function builds a
GenomicState object which you can then use with
derfinder::annotateRegions(). This information is then used by packages
like derfinderPlot." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicState</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.11</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/GenomicState.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LieberInstitute/GenomicState/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Build a GenomicState object for Gencode TxDb objects</h1>
    <small class="dont-index">Source: <a href='https://github.com/LieberInstitute/GenomicState/blob/master/R/gencode_genomic_state.R'><code>R/gencode_genomic_state.R</code></a></small>
    <div class="hidden name"><code>gencode_genomic_state.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Based on a <code>TxDb</code> object built by <code><a href='gencode_txdb.html'>gencode_txdb()</a></code> this function builds a
<code>GenomicState</code> object which you can then use with
<code><a href='https://rdrr.io/pkg/derfinder/man/annotateRegions.html'>derfinder::annotateRegions()</a></code>. This information is then used by packages
like <code>derfinderPlot</code>.</p>
    </div>

    <pre class="usage"><span class='fu'>gencode_genomic_state</span>(<span class='no'>txdb</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>txdb</th>
      <td><p>A <a href='https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html'>GenomicFeatures::TxDb</a> object built with
<code><a href='gencode_txdb.html'>gencode_txdb()</a></code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A GenomicState object with the gene symbols as built using
<code><a href='https://rdrr.io/pkg/derfinder/man/makeGenomicState.html'>derfinder::makeGenomicState()</a></code>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Note that not all genes will have symbols as many will be <code>NA</code>.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Based on code for the <code>brainflowprobes</code> package at:
<a href='https://github.com/LieberInstitute/brainflowprobes/blob/master/data-raw/create_sysdata.R'>https://github.com/LieberInstitute/brainflowprobes/blob/master/data-raw/create_sysdata.R</a></p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='gencode_txdb.html'>gencode_txdb()</a></code> <code><a href='gencode_annotated_genes.html'>gencode_annotated_genes()</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>## Start from scratch if you want:</span>
<span class='kw'>if</span> (<span class='fl'>FALSE</span>) {
<span class='no'>txdb_v31_hg19_chr21</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='gencode_txdb.html'>gencode_txdb</a></span>(<span class='st'>"31"</span>, <span class='st'>"hg19"</span>, <span class='kw'>chrs</span> <span class='kw'>=</span> <span class='st'>"chr21"</span>)
}

<span class='co'>## or read in the txdb object for hg19 chr21 from this package</span>
<span class='no'>txdb_v31_hg19_chr21</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>AnnotationDbi</span><span class='kw ns'>::</span><span class='fu'><a href='https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html'>loadDb</a></span>(
    <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='st'>"txdb_v31_hg19_chr21.sqlite"</span>,
        <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"GenomicState"</span>
    )
)

<span class='co'>## Now build the GenomicState object</span>
<span class='no'>gs_v31_hg19_chr21</span> <span class='kw'>&lt;-</span> <span class='fu'>gencode_genomic_state</span>(<span class='no'>txdb_v31_hg19_chr21</span>)</div><div class='output co'>#&gt; <span class='message'>2020-04-24 03:59:06 making the GenomicState object</span></div><div class='output co'>#&gt; <span class='message'>extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:1 mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>2020-04-24 03:59:13 finding gene symbols</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>2020-04-24 03:59:13 adding gene symbols to the GenomicState</span></div><div class='input'>
<span class='co'>## Explore the result</span>
<span class='no'>gs_v31_hg19_chr21</span></div><div class='output co'>#&gt; $fullGenome
#&gt; GRanges object with 7871 ranges and 5 metadata columns:
#&gt;        seqnames            ranges strand |   theRegion         tx_id
#&gt;           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;IntegerList&gt;
#&gt;      1    chr21   9492380-9492817      + |        exon             1
#&gt;      2    chr21   9590294-9590395      + |        exon             2
#&gt;      3    chr21   9647910-9648694      + |        exon           3,4
#&gt;      4    chr21   9648695-9650108      + |      intron           3,4
#&gt;      5    chr21   9650109-9650168      + |        exon           3,4
#&gt;    ...      ...               ...    ... .         ...           ...
#&gt;   7867    chr21 47865683-47878803      * |  intergenic              
#&gt;   7868    chr21 47989929-48018516      * |  intergenic              
#&gt;   7869    chr21 48025122-48055506      * |  intergenic              
#&gt;   7870    chr21 48085037-48110675      * |  intergenic              
#&gt;   7871    chr21 48111139-48129895      * |  intergenic              
#&gt;                                        tx_name          gene          symbol
#&gt;                                &lt;CharacterList&gt; &lt;IntegerList&gt; &lt;CharacterList&gt;
#&gt;      1                     ENST00000625020.1_1           776              NA
#&gt;      2                     ENST00000625098.1_1           754              NA
#&gt;      3 ENST00000623794.3_1,ENST00000624813.1_1           766    LOC101930100
#&gt;      4 ENST00000623794.3_1,ENST00000624813.1_1           766    LOC101930100
#&gt;      5 ENST00000623794.3_1,ENST00000624813.1_1           766    LOC101930100
#&gt;    ...                                     ...           ...             ...
#&gt;   7867                                                                      
#&gt;   7868                                                                      
#&gt;   7869                                                                      
#&gt;   7870                                                                      
#&gt;   7871                                                                      
#&gt;   -------
#&gt;   seqinfo: 1 sequence from hg19 genome
#&gt; 
#&gt; $codingGenome
#&gt; GRanges object with 10361 ranges and 5 metadata columns:
#&gt;         seqnames            ranges strand |   theRegion         tx_id
#&gt;            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;IntegerList&gt;
#&gt;       1    chr21   9490380-9492379      + |    promoter             1
#&gt;       2    chr21   9492380-9492817      + |        exon             1
#&gt;       3    chr21   9588294-9590293      + |    promoter             2
#&gt;       4    chr21   9590294-9590395      + |        exon             2
#&gt;       5    chr21   9590396-9590493      + |    promoter             2
#&gt;     ...      ...               ...    ... .         ...           ...
#&gt;   10357    chr21 47865683-47876803      * |  intergenic              
#&gt;   10358    chr21 47989929-48018516      * |  intergenic              
#&gt;   10359    chr21 48027122-48053506      * |  intergenic              
#&gt;   10360    chr21 48085037-48108675      * |  intergenic              
#&gt;   10361    chr21 48111139-48129895      * |  intergenic              
#&gt;                     tx_name          gene          symbol
#&gt;             &lt;CharacterList&gt; &lt;IntegerList&gt; &lt;CharacterList&gt;
#&gt;       1 ENST00000625020.1_1           776              NA
#&gt;       2 ENST00000625020.1_1           776              NA
#&gt;       3 ENST00000625098.1_1           754              NA
#&gt;       4 ENST00000625098.1_1           754              NA
#&gt;       5 ENST00000625098.1_1           754              NA
#&gt;     ...                 ...           ...             ...
#&gt;   10357                                                  
#&gt;   10358                                                  
#&gt;   10359                                                  
#&gt;   10360                                                  
#&gt;   10361                                                  
#&gt;   -------
#&gt;   seqinfo: 1 sequence from hg19 genome
#&gt; </div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


