<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Access GenomicState objects through AnnotationHub — GenomicStateHub • GenomicState</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Access GenomicState objects through AnnotationHub — GenomicStateHub" />
<meta property="og:description" content="This function uses AnnotationHub to obtain the objects provided by this
package. These are: the TxDb object made with gencode_txdb(), the
annotated genes object made with gencode_annotated_genes() or the
GenomicState object made with gencode_genomic_state()." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicState</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.11</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/GenomicState.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LieberInstitute/GenomicState/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Access GenomicState objects through AnnotationHub</h1>
    <small class="dont-index">Source: <a href='https://github.com/LieberInstitute/GenomicState/blob/master/R/GenomicStateHub.R'><code>R/GenomicStateHub.R</code></a></small>
    <div class="hidden name"><code>GenomicStateHub.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function uses <code>AnnotationHub</code> to obtain the objects provided by this
package. These are: the <code>TxDb</code> object made with <code><a href='gencode_txdb.html'>gencode_txdb()</a></code>, the
annotated genes object made with <code><a href='gencode_annotated_genes.html'>gencode_annotated_genes()</a></code> or the
GenomicState object made with <code><a href='gencode_genomic_state.html'>gencode_genomic_state()</a></code>.</p>
    </div>

    <pre class="usage"><span class='fu'>GenomicStateHub</span>(
  <span class='kw'>version</span> <span class='kw'>=</span> <span class='st'>"31"</span>,
  <span class='kw'>genome</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"hg38"</span>, <span class='st'>"hg19"</span>),
  <span class='kw'>filetype</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"TxDb"</span>, <span class='st'>"AnnotatedGenes"</span>, <span class='st'>"GenomicState"</span>),
  <span class='kw'>ah</span> <span class='kw'>=</span> <span class='kw pkg'>AnnotationHub</span><span class='kw ns'>::</span><span class='fu'><a href='https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html'>AnnotationHub</a></span>()
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>version</th>
      <td><p>A <code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code> with the Gencode version number.</p></td>
    </tr>
    <tr>
      <th>genome</th>
      <td><p>A <code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code> with the human genome version number. Valid
options are <code>'hg38'</code> or <code>'hg19'</code>.</p></td>
    </tr>
    <tr>
      <th>filetype</th>
      <td><p>A <code><a href='https://rdrr.io/r/base/character.html'>character()</a></code> with either <code>TxDb</code>, <code>AnnotatedGenes</code> or
<code>GenomicState</code>.</p></td>
    </tr>
    <tr>
      <th>ah</th>
      <td><p>An <code>AnnotationHub</code> object
<a href='https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html'>AnnotationHub-class</a>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>The <a href='https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html'>AnnotationHub-class</a> query
for the file you requested.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='gencode_txdb.html'>gencode_txdb()</a></code> <code><a href='gencode_annotated_genes.html'>gencode_annotated_genes()</a></code>
<code><a href='gencode_genomic_state.html'>gencode_genomic_state()</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>## Query AnnotationHub for the GenomicState object for Gencode v31 on</span>
<span class='co'>## hg19 coordinates</span>
<span class='no'>hub_query_gs_gencode_v31_hg19</span> <span class='kw'>&lt;-</span> <span class='fu'>GenomicStateHub</span>(
    <span class='kw'>version</span> <span class='kw'>=</span> <span class='st'>"31"</span>,
    <span class='kw'>genome</span> <span class='kw'>=</span> <span class='st'>"hg19"</span>,
    <span class='kw'>filetype</span> <span class='kw'>=</span> <span class='st'>"GenomicState"</span>
)</div><div class='output co'>#&gt; <span class='message'>using temporary cache /tmp/RtmpjZoPt2/BiocFileCache</span></div><div class='output co'>#&gt; <span class='message'>snapshotDate(): 2020-03-31</span></div><div class='input'><span class='no'>hub_query_gs_gencode_v31_hg19</span></div><div class='output co'>#&gt; AnnotationHub with 1 record
#&gt; # snapshotDate(): 2020-03-31
#&gt; # names(): AH75184
#&gt; # $dataprovider: GENCODE
#&gt; # $species: Homo sapiens
#&gt; # $rdataclass: list
#&gt; # $rdatadateadded: 2019-10-22
#&gt; # $title: GenomicState for Gencode v31 on hg19 coordinates
#&gt; # $description: Gencode v31 GenomicState from derfinder::makeGenomicState() ...
#&gt; # $taxonomyid: 9606
#&gt; # $genome: GRCh37
#&gt; # $sourcetype: GTF
#&gt; # $sourceurl: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/releas...
#&gt; # $sourcesize: NA
#&gt; # $tags: c("Gencode", "GenomicState", "hg19", "v31") 
#&gt; # retrieve record with 'object[["AH75184"]]' </div><div class='input'>

<span class='co'>## Check the metadata</span>
<span class='fu'>mcols</span>(<span class='no'>hub_query_gs_gencode_v31_hg19</span>)</div><div class='output co'>#&gt; DataFrame with 1 row and 15 columns
#&gt;                                                    title dataprovider
#&gt;                                              &lt;character&gt;  &lt;character&gt;
#&gt; AH75184 GenomicState for Gencode v31 on hg19 coordinates      GENCODE
#&gt;              species taxonomyid      genome
#&gt;          &lt;character&gt;  &lt;integer&gt; &lt;character&gt;
#&gt; AH75184 Homo sapiens       9606      GRCh37
#&gt;                                                                                                                                                                                                     description
#&gt;                                                                                                                                                                                                     &lt;character&gt;
#&gt; AH75184 Gencode v31 GenomicState from derfinder::makeGenomicState() on hg19 coordinates. This is useful for packages such as derfinder and derfinderPlot. For more information, check the GenomicState package.
#&gt;         coordinate_1_based                                      maintainer
#&gt;                  &lt;integer&gt;                                     &lt;character&gt;
#&gt; AH75184                  1 Leonardo Collado-Torres &lt;lcolladotor@gmail.com&gt;
#&gt;         rdatadateadded preparerclass                          tags  rdataclass
#&gt;            &lt;character&gt;   &lt;character&gt;                        &lt;list&gt; &lt;character&gt;
#&gt; AH75184     2019-10-22  GenomicState Gencode,GenomicState,hg19,...        list
#&gt;                                                      rdatapath
#&gt;                                                    &lt;character&gt;
#&gt; AH75184 GenomicState/gencode/gencode_v31_hg19_GenomicState.rda
#&gt;                                                                                                                     sourceurl
#&gt;                                                                                                                   &lt;character&gt;
#&gt; AH75184 ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_31/GRCh37_mapping/gencode.v31lift37.annotation.gtf.gz
#&gt;          sourcetype
#&gt;         &lt;character&gt;
#&gt; AH75184         GTF</div><div class='input'>
<span class='co'>## Access the file through AnnotationHub</span>
<span class='kw'>if</span> (<span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>hub_query_gs_gencode_v31_hg19</span>) <span class='kw'>==</span> <span class='fl'>1</span>) {
    <span class='no'>hub_gs_gencode_v31_hg19</span> <span class='kw'>&lt;-</span> <span class='no'>hub_query_gs_gencode_v31_hg19</span><span class='kw'>[[</span><span class='fl'>1</span>]]

    <span class='no'>hub_gs_gencode_v31_hg19</span>
}</div><div class='output co'>#&gt; <span class='message'>downloading 1 resources</span></div><div class='output co'>#&gt; <span class='message'>retrieving 1 resource</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; $fullGenome</div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: IRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='output co'>#&gt; GRanges object with 659263 ranges and 5 metadata columns:
#&gt;          seqnames            ranges strand |   theRegion         tx_id
#&gt;             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;IntegerList&gt;
#&gt;        1     chr1       11869-12227      + |        exon           1,2
#&gt;        2     chr1       12228-12612      + |      intron           1,2
#&gt;        3     chr1       12613-12721      + |        exon           1,2
#&gt;        4     chr1       12722-12974      + |      intron           1,2
#&gt;        5     chr1       12975-13052      + |        exon             2
#&gt;      ...      ...               ...    ... .         ...           ...
#&gt;   659259     chrY 59208555-59214013      * |  intergenic              
#&gt;   659260     chrY 59276440-59311662      * |  intergenic              
#&gt;   659261     chrY 59311997-59318040      * |  intergenic              
#&gt;   659262     chrY 59318921-59330251      * |  intergenic              
#&gt;   659263     chrY 59360549-59373566      * |  intergenic              
#&gt;                                          tx_name          gene          symbol
#&gt;                                  &lt;CharacterList&gt; &lt;IntegerList&gt; &lt;CharacterList&gt;
#&gt;        1 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1
#&gt;        2 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1
#&gt;        3 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1
#&gt;        4 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1
#&gt;        5                     ENST00000450305.2_1         26085         DDX11L1
#&gt;      ...                                     ...           ...             ...
#&gt;   659259                                                                      
#&gt;   659260                                                                      
#&gt;   659261                                                                      
#&gt;   659262                                                                      
#&gt;   659263                                                                      
#&gt;   -------
#&gt;   seqinfo: 24 sequences from hg19 genome
#&gt; 
#&gt; $codingGenome
#&gt; GRanges object with 878954 ranges and 5 metadata columns:
#&gt;          seqnames            ranges strand |   theRegion         tx_id
#&gt;             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;IntegerList&gt;
#&gt;        1     chr1        9869-11868      + |    promoter           1,2
#&gt;        2     chr1       11869-12227      + |        exon           1,2
#&gt;        3     chr1       12228-12612      + |      intron           1,2
#&gt;        4     chr1       12613-12721      + |        exon           1,2
#&gt;        5     chr1       12722-12974      + |      intron           1,2
#&gt;      ...      ...               ...    ... .         ...           ...
#&gt;   878950     chrY 59208555-59212013      * |  intergenic              
#&gt;   878951     chrY 59276440-59311662      * |  intergenic              
#&gt;   878952     chrY 59313997-59318040      * |  intergenic              
#&gt;   878953     chrY 59320921-59328251      * |  intergenic              
#&gt;   878954     chrY 59362549-59373566      * |  intergenic              
#&gt;                                          tx_name          gene          symbol
#&gt;                                  &lt;CharacterList&gt; &lt;IntegerList&gt; &lt;CharacterList&gt;
#&gt;        1 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1
#&gt;        2 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1
#&gt;        3 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1
#&gt;        4 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1
#&gt;        5 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1
#&gt;      ...                                     ...           ...             ...
#&gt;   878950                                                                      
#&gt;   878951                                                                      
#&gt;   878952                                                                      
#&gt;   878953                                                                      
#&gt;   878954                                                                      
#&gt;   -------
#&gt;   seqinfo: 24 sequences from hg19 genome
#&gt; </div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


