<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>GenomicState • GenomicState</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="GenomicState">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicState</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/GenomicState.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/GenomicState">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>GenomicState</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/GenomicState/blob/master/vignettes/GenomicState.Rmd"><code>vignettes/GenomicState.Rmd</code></a></small>
      <div class="hidden name"><code>GenomicState.Rmd</code></div>

    </div>

    
    
<div id="basics" class="section level1">
<h1 class="hasAnchor">
<a href="#basics" class="anchor"></a>Basics</h1>
<div id="install-genomicstate" class="section level2">
<h2 class="hasAnchor">
<a href="#install-genomicstate" class="anchor"></a>Install <em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em>
</h2>
<p><code>R</code> is an open-source statistical environment which can be easily modified to enhance its functionality via packages. <em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em> is a <code>R</code> package available via GitHub. <code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/">CRAN</a> after which you can install <em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em> by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co">## If necessary</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">'remotes'</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">## Install GenomicState</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5">remotes<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/remotes/man/install_github.html">install_github</a></span>(<span class="st">'LieberInstitute/GenomicState'</span>)</a></code></pre></div>
</div>
<div id="required-knowledge" class="section level2">
<h2 class="hasAnchor">
<a href="#required-knowledge" class="anchor"></a>Required knowledge</h2>
<p><em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em> is based on many other packages and in particular in those that have implemented the infrastructure needed for dealing with annotation data. That is, packages like <em><a href="https://bioconductor.org/packages/3.9/rtracklayer">rtracklayer</a></em> that allow you to import the data. A <em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em> user is not expected to deal with those packages directly but will need to be familiar with <em><a href="https://bioconductor.org/packages/3.9/derfinder">derfinder</a></em> and <em><a href="https://bioconductor.org/packages/3.9/derfinderPlot">derfinderPlot</a></em> to understand the results <em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em> generates.</p>
<p>If you are asking yourself the question “Where do I start using Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU">this blog post</a>.</p>
</div>
<div id="asking-for-help" class="section level2">
<h2 class="hasAnchor">
<a href="#asking-for-help" class="anchor"></a>Asking for help</h2>
<p>As package developers, we try to explain clearly how to use our packages and in which order to use the functions. But <code>R</code> and <code>Bioconductor</code> have a steep learning curve so it is critical to learn where to ask for help. The blog post quoted above mentions some but we would like to highlight the <a href="https://support.bioconductor.org/">Bioconductor support site</a> as the main resource for getting help regarding Bioconductor. Other alternatives are available such as creating GitHub issues and tweeting. However, please note that if you want to receive help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/">posting guidelines</a>. It is particularly critical that you provide a small reproducible example and your session information so package developers can track down the source of the error.</p>
</div>
<div id="citing-genomicstate" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-genomicstate" class="anchor"></a>Citing <em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em>
</h2>
<p>We hope that <em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co">## Citation info</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span>(<span class="st">'GenomicState'</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">#&gt; To cite package 'GenomicState' in publications use:</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">#&gt;   Leonardo Collado-Torres (2019). GenomicState: Build and access</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co">#&gt;   GenomicState objects for use with derfinder tools from sources</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">#&gt;   like Gencode. R package version 0.99.0.</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co">#&gt;   https://github.com/LieberInstitute/GenomicState</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="co">#&gt;   @Manual{,</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="co">#&gt;     title = {GenomicState: Build and access GenomicState objects for use with derfinder tools from</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co">#&gt; sources like Gencode},</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="co">#&gt;     author = {Leonardo Collado-Torres},</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">#&gt;     year = {2019},</span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="co">#&gt;     note = {R package version 0.99.0},</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"><span class="co">#&gt;     url = {https://github.com/LieberInstitute/GenomicState},</span></a>
<a class="sourceLine" id="cb2-20" data-line-number="20"><span class="co">#&gt;   }</span></a></code></pre></div>
</div>
</div>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>The <em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em> package was developed for speeding up analyses that require these objects and in particular those that rely on Gencode annotation data. The package <em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em> provides functions for building <code>GenomicState</code> objects from diverse annotation sources such as <a href="https://www.gencodegenes.org/human/releases.html"><code>Gencode</code></a>. It also provides a way to load pre-computed <code>GenomicState</code> objects if you are working at <a href="https://jhpce.jhu.edu/">JHPCE</a>. These <code>GenomicState</code> objects are normally created using <a href="https://rdrr.io/bioc/derfinder/man/makeGenomicState.html">derfinder::makeGenomicState()</a> and can be used for annotating regions with <a href="https://rdrr.io/bioc/derfinder/man/annotateRegions.html">derfinder::annotateRegions()</a> which are in turn used by <a href="https://rdrr.io/bioc/derfinderPlot/man/plotRegionCoverage.html">derfinderPlot::plotRegionCoverage()</a>.</p>
<p>To get started, load the <em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em> package.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'GenomicState'</span>)</a></code></pre></div>
</div>
<div id="jhpce" class="section level1">
<h1 class="hasAnchor">
<a href="#jhpce" class="anchor"></a>JHPCE</h1>
<p>TODO</p>
</div>
<div id="build-objects" class="section level1">
<h1 class="hasAnchor">
<a href="#build-objects" class="anchor"></a>Build objects</h1>
<p>TODO</p>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>The <em><a href="https://github.com/LieberInstitute/GenomicState">GenomicState</a></em> package <a id="cite-Collado-Torres_2019"></a>(<a href="https://github.com/LieberInstitute/GenomicState">Collado-Torres, 2019</a>) was made possible thanks to:</p>
<ul>
<li>R <a id="cite-R_Core_Team_2019"></a>(<a href="https://www.R-project.org/">R Core Team, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.9/BiocStyle">BiocStyle</a></em> <a id="cite-Oles_2019"></a>(<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.9/GenomeInfoDb">GenomeInfoDb</a></em> <a id="cite-GenomeInfoDb"></a>(<a href="https://doi.org/10.18129/B9.bioc.GenomeInfoDb">Arora, Morgan, Carlson, and Pagès, 2017</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> <a id="cite-Boettiger_2019"></a>(<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> <a id="cite-Xie_2014"></a>(<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em></li>
<li>
<em><a href="https://bioconductor.org/packages/3.9/rtracklayer">rtracklayer</a></em> <a id="cite-Lawrence_2009"></a>(<a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract">Lawrence, Gentleman, and Carey, 2009</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo">sessioninfo</a></em> <a id="cite-Csardi_2018"></a>(<a href="https://CRAN.R-project.org/package=sessioninfo">Csárdi, core, Wickham, Chang, et al., 2018</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat">testthat</a></em> <a id="cite-Wickham_2011"></a>(<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf">Wickham, 2011</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.9/GenomicFeatures">GenomicFeatures</a></em> <a id="cite-Lawrence_2013"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.9/bumphunter">bumphunter</a></em></li>
<li>
<em><a href="https://bioconductor.org/packages/3.9/derfinder">derfinder</a></em> <a id="cite-Collado-Torres_2017"></a>(<a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852">Collado-Torres, Nellore, Frazee, Wilks, et al., 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.9/AnnotationDbi">AnnotationDbi</a></em> <a id="cite-Pages_2019"></a>(<a href="#bib-Pages_2019">Pagès, Carlson, Falcon, and Li, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.9/IRanges">IRanges</a></em> <a id="cite-Lawrence_2013a"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.9/org.Hs.eg.db">org.Hs.eg.db</a></em> <a id="cite-Carlson_2019"></a>(<a href="#bib-Carlson_2019">Carlson, 2019</a>)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co">## Create the vignette</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'rmarkdown'</span>)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">'GenomicState.Rmd'</span>))</a>
<a class="sourceLine" id="cb4-4" data-line-number="4"></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">## Extract the R code</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'knitr'</span>)</a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="kw"><a href="https://rdrr.io/pkg/knitr/man/knit.html">knit</a></span>(<span class="st">'GenomicState.Rmd'</span>, <span class="dt">tangle =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code>#&gt; [1] "2019-10-03 12:49:40 EDT"</code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code>#&gt; Time difference of 2.243 secs</code></pre>
<p><code>R</code> session information.</p>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 3.6.1 (2019-07-05)
#&gt;  os       macOS Mojave 10.14.6        
#&gt;  system   x86_64, darwin15.6.0        
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_US.UTF-8                 
#&gt;  ctype    en_US.UTF-8                 
#&gt;  tz       America/New_York            
#&gt;  date     2019-10-03                  
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package       * version  date       lib source        
#&gt;  assertthat      0.2.1    2019-03-21 [2] CRAN (R 3.6.0)
#&gt;  backports       1.1.4    2019-04-10 [2] CRAN (R 3.6.0)
#&gt;  bibtex          0.4.2    2017-06-30 [2] CRAN (R 3.6.0)
#&gt;  BiocManager     1.30.4   2018-11-13 [2] CRAN (R 3.6.0)
#&gt;  BiocStyle     * 2.12.0   2019-05-02 [2] Bioconductor  
#&gt;  cli             1.1.0    2019-03-19 [2] CRAN (R 3.6.0)
#&gt;  crayon          1.3.4    2017-09-16 [2] CRAN (R 3.6.0)
#&gt;  desc            1.2.0    2018-05-01 [2] CRAN (R 3.6.0)
#&gt;  digest          0.6.21   2019-09-20 [2] CRAN (R 3.6.0)
#&gt;  evaluate        0.14     2019-05-28 [2] CRAN (R 3.6.0)
#&gt;  fs              1.3.1    2019-05-06 [2] CRAN (R 3.6.0)
#&gt;  GenomicState  * 0.99.0   2019-10-03 [1] Bioconductor  
#&gt;  htmltools       0.3.6    2017-04-28 [2] CRAN (R 3.6.0)
#&gt;  httr            1.4.1    2019-08-05 [2] CRAN (R 3.6.0)
#&gt;  jsonlite        1.6      2018-12-07 [2] CRAN (R 3.6.0)
#&gt;  knitcitations * 1.0.10   2019-09-15 [2] CRAN (R 3.6.0)
#&gt;  knitr           1.25     2019-09-18 [2] CRAN (R 3.6.0)
#&gt;  lubridate       1.7.4    2018-04-11 [2] CRAN (R 3.6.0)
#&gt;  magrittr        1.5      2014-11-22 [2] CRAN (R 3.6.0)
#&gt;  MASS            7.3-51.4 2019-03-31 [2] CRAN (R 3.6.1)
#&gt;  memoise         1.1.0    2017-04-21 [2] CRAN (R 3.6.0)
#&gt;  pkgdown         1.4.1    2019-09-15 [2] CRAN (R 3.6.0)
#&gt;  plyr            1.8.4    2016-06-08 [2] CRAN (R 3.6.0)
#&gt;  R6              2.4.0    2019-02-14 [2] CRAN (R 3.6.0)
#&gt;  Rcpp            1.0.2    2019-07-25 [2] CRAN (R 3.6.0)
#&gt;  RefManageR      1.2.12   2019-04-03 [2] CRAN (R 3.6.0)
#&gt;  rlang           0.4.0    2019-06-25 [2] CRAN (R 3.6.0)
#&gt;  rmarkdown       1.15     2019-08-21 [2] CRAN (R 3.6.1)
#&gt;  rprojroot       1.3-2    2018-01-03 [2] CRAN (R 3.6.0)
#&gt;  rstudioapi      0.10     2019-03-19 [2] CRAN (R 3.6.0)
#&gt;  sessioninfo   * 1.1.1    2018-11-05 [2] CRAN (R 3.6.0)
#&gt;  stringi         1.4.3    2019-03-12 [2] CRAN (R 3.6.0)
#&gt;  stringr         1.4.0    2019-02-10 [2] CRAN (R 3.6.0)
#&gt;  withr           2.1.2    2018-03-15 [2] CRAN (R 3.6.0)
#&gt;  xfun            0.9      2019-08-21 [2] CRAN (R 3.6.0)
#&gt;  xml2            1.2.2    2019-08-09 [2] CRAN (R 3.6.0)
#&gt;  yaml            2.2.0    2018-07-25 [2] CRAN (R 3.6.0)
#&gt; 
#&gt; [1] /private/var/folders/cx/n9s558kx6fb7jf5z_pgszgb80000gn/T/Rtmpis7rxK/temp_libpath341f3f7f28ae
#&gt; [2] /Library/Frameworks/R.framework/Versions/3.6/Resources/library</code></pre>
</div>
<div id="bibliography" class="section level1">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.9/BiocStyle">BiocStyle</a></em> (<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2019</a>), <em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>) and <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> running behind the scenes.</p>
<p>Citations were made with <em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> (<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>).</p>
<p>[1] S. Arora, M. Morgan, M. Carlson, and H. Pagès. <em>GenomeInfoDb: Utilities for manipulating chromosome and other ‘seqname’ identifiers</em>. 2017. DOI: 10.18129/B9.bioc.GenomeInfoDb.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for ‘Knitr’ Markdown Files</em>. R package version 1.0.10. 2019. &lt;URL: <a href="https://CRAN.R-project.org/package=knitcitations" class="uri">https://CRAN.R-project.org/package=knitcitations</a>&gt;.</p>
<p>[3] M. Carlson. <em>org.Hs.eg.db: Genome wide annotation for Human</em>. R package version 3.8.2. 2019.</p>
<p>[4] L. Collado-Torres. <em>GenomicState: Build and access GenomicState objects for use with derfinder tools from sources like Gencode</em>. R package version 0.99.0. 2019. &lt;URL: <a href="https://github.com/LieberInstitute/GenomicState" class="uri">https://github.com/LieberInstitute/GenomicState</a>&gt;.</p>
<p>[5] L. Collado-Torres, A. Nellore, A. C. Frazee, C. Wilks, et al. “Flexible expressed region analysis for RNA-seq with derfinder”. In: <em>Nucl. Acids Res.</em> (2017). DOI: 10.1093/nar/gkw852. &lt;URL: <a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852" class="uri">http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852</a>&gt;.</p>
<p>[6] G. Csárdi, R. core, H. Wickham, W. Chang, et al. <em>sessioninfo: R Session Information</em>. R package version 1.1.1. 2018. &lt;URL: <a href="https://CRAN.R-project.org/package=sessioninfo" class="uri">https://CRAN.R-project.org/package=sessioninfo</a>&gt;.</p>
<p>[7] M. Lawrence, R. Gentleman, and V. Carey. “rtracklayer: an R package for interfacing with genome browsers”. In: <em>Bioinformatics</em> 25 (2009), pp. 1841-1842. DOI: 10.1093/bioinformatics/btp328. &lt;URL: <a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract" class="uri">http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract</a>&gt;.</p>
<p>[8] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[9] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[10] A. Oleś, M. Morgan, and W. Huber. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.12.0. 2019. &lt;URL: <a href="https://github.com/Bioconductor/BiocStyle" class="uri">https://github.com/Bioconductor/BiocStyle</a>&gt;.</p>
<p>[11] H. Pagès, M. Carlson, S. Falcon, and N. Li. <em>AnnotationDbi: Manipulation of SQLite-based annotations in Bioconductor</em>. R package version 1.46.1. 2019.</p>
<p>[12] R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2019. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[13] H. Wickham. “testthat: Get Started with Testing”. In: <em>The R Journal</em> 3 (2011), pp. 5-10. &lt;URL: <a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="uri">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>&gt;.</p>
<p>[14] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V. Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and Hall/CRC, 2014. &lt;URL: <a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#basics">Basics</a><ul class="nav nav-pills nav-stacked">
<li><a href="#install-genomicstate">Install <em><span>GenomicState</span></em></a></li>
      <li><a href="#required-knowledge">Required knowledge</a></li>
      <li><a href="#asking-for-help">Asking for help</a></li>
      <li><a href="#citing-genomicstate">Citing <em><span>GenomicState</span></em></a></li>
      </ul>
</li>
      <li><a href="#overview">Overview</a></li>
      <li><a href="#jhpce">JHPCE</a></li>
      <li><a href="#build-objects">Build objects</a></li>
      <li><a href="#reproducibility">Reproducibility</a></li>
      <li><a href="#bibliography">Bibliography</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
