<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to GenomicState • GenomicState</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to GenomicState">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicState</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/GenomicState.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/GenomicState">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction to GenomicState</h1>
                        <h4 class="author">Leonardo Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 class="date">22 October 2019</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/GenomicState/blob/master/vignettes/GenomicState.Rmd"><code>vignettes/GenomicState.Rmd</code></a></small>
      <div class="hidden name"><code>GenomicState.Rmd</code></div>

    </div>

    
    
<div id="basics" class="section level1">
<h1 class="hasAnchor">
<a href="#basics" class="anchor"></a>Basics</h1>
<div id="install-genomicstate" class="section level2">
<h2 class="hasAnchor">
<a href="#install-genomicstate" class="anchor"></a>Install <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em>
</h2>
<p><code>R</code> is an open-source statistical environment which can be easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> is a <code>R</code> package available via GitHub. <code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/">CRAN</a> after which you can install <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))
    <span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)

BiocManager<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"GenomicState"</span>)

## Check that you have a valid Bioconductor installation
BiocManager<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html">valid</a></span>()</code></pre></div>
</div>
<div id="required-knowledge" class="section level2">
<h2 class="hasAnchor">
<a href="#required-knowledge" class="anchor"></a>Required knowledge</h2>
<p><em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> is based on many other packages and in particular in those that have implemented the infrastructure needed for dealing with annotation data. That is, packages like <em><a href="https://bioconductor.org/packages/3.10/rtracklayer">rtracklayer</a></em> that allow you to import the data. A <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> user is not expected to deal with those packages directly but will need to be familiar with <em><a href="https://bioconductor.org/packages/3.10/derfinder">derfinder</a></em> and <em><a href="https://bioconductor.org/packages/3.10/derfinderPlot">derfinderPlot</a></em> to understand the results <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> generates. Furthermore, it’ll be useful for the user to know the syntax of <em><a href="https://bioconductor.org/packages/3.10/AnnotationHub">AnnotationHub</a></em> <a id="cite-Morgan_2019"></a>(<a href="#bib-Morgan_2019">Morgan, 2019</a>) in order to query and load the data provided by this package.</p>
<p>If you are asking yourself the question “Where do I start using Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU">this blog post</a>.</p>
</div>
<div id="asking-for-help" class="section level2">
<h2 class="hasAnchor">
<a href="#asking-for-help" class="anchor"></a>Asking for help</h2>
<p>As package developers, we try to explain clearly how to use our packages and in which order to use the functions. But <code>R</code> and <code>Bioconductor</code> have a steep learning curve so it is critical to learn where to ask for help. The blog post quoted above mentions some but we would like to highlight the <a href="https://support.bioconductor.org/">Bioconductor support site</a> as the main resource for getting help regarding Bioconductor. Other alternatives are available such as creating GitHub issues and tweeting. However, please note that if you want to receive help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/">posting guidelines</a>. It is particularly critical that you provide a small reproducible example and your session information so package developers can track down the source of the error.</p>
</div>
<div id="citing-genomicstate" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-genomicstate" class="anchor"></a>Citing <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Citation info
<span class="kw"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span>(<span class="st">'GenomicState'</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; To cite package 'GenomicState' in publications use:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   Leonardo Collado-Torres (2019). GenomicState: Build and access</span>
<span class="co">#&gt;   GenomicState objects for use with derfinder tools from sources</span>
<span class="co">#&gt;   like Gencode. R package version 0.99.5.</span>
<span class="co">#&gt;   https://github.com/LieberInstitute/GenomicState</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; A BibTeX entry for LaTeX users is</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   @Manual{,</span>
<span class="co">#&gt;     title = {GenomicState: Build and access GenomicState objects for use with derfinder tools from</span>
<span class="co">#&gt; sources like Gencode},</span>
<span class="co">#&gt;     author = {Leonardo Collado-Torres},</span>
<span class="co">#&gt;     year = {2019},</span>
<span class="co">#&gt;     note = {R package version 0.99.5},</span>
<span class="co">#&gt;     url = {https://github.com/LieberInstitute/GenomicState},</span>
<span class="co">#&gt;   }</span></code></pre></div>
</div>
</div>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> package was developed for speeding up analyses that require these objects and in particular those that rely on Gencode annotation data. The package <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> provides functions for building <code>GenomicState</code> objects from diverse annotation sources such as <a href="https://www.gencodegenes.org/human/releases.html"><code>Gencode</code></a>. It also provides a way to load pre-computed <code>GenomicState</code> objects if you are working at <a href="https://jhpce.jhu.edu/">JHPCE</a>. These <code>GenomicState</code> objects are normally created using <a href="https://rdrr.io/bioc/derfinder/man/makeGenomicState.html">derfinder::makeGenomicState()</a> and can be used for annotating regions with <a href="https://rdrr.io/bioc/derfinder/man/annotateRegions.html">derfinder::annotateRegions()</a> which are in turn used by <a href="https://rdrr.io/bioc/derfinderPlot/man/plotRegionCoverage.html">derfinderPlot::plotRegionCoverage()</a>.</p>
<p>To get started, load the <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'GenomicState'</span>)</code></pre></div>
</div>
<div id="annotationhub" class="section level1">
<h1 class="hasAnchor">
<a href="#annotationhub" class="anchor"></a>AnnotationHub</h1>
<p>Using the <code>GencodeStateHub()</code> function you can query and download the data from <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> using <em><a href="https://bioconductor.org/packages/3.10/AnnotationHub">AnnotationHub</a></em> (<a href="#bib-Morgan_2019">Morgan, 2019</a>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Query AnnotationHub for the GenomicState object for Gencode v31 on
## hg19 coordinates
hub_query_gs_gencode_v31_hg19 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/GenomicStateHub.html">GenomicStateHub</a></span>(<span class="dt">version =</span> <span class="st">'31'</span>,
    <span class="dt">genome =</span> <span class="st">'hg19'</span>,
    <span class="dt">filetype =</span> <span class="st">'GenomicState'</span>)
<span class="co">#&gt; snapshotDate(): 2019-10-22</span>
hub_query_gs_gencode_v31_hg19
<span class="co">#&gt; AnnotationHub with 1 record</span>
<span class="co">#&gt; # snapshotDate(): 2019-10-22 </span>
<span class="co">#&gt; # names(): AH75184</span>
<span class="co">#&gt; # $dataprovider: GENCODE</span>
<span class="co">#&gt; # $species: Homo sapiens</span>
<span class="co">#&gt; # $rdataclass: list</span>
<span class="co">#&gt; # $rdatadateadded: 2019-10-22</span>
<span class="co">#&gt; # $title: GenomicState for Gencode v31 on hg19 coordinates</span>
<span class="co">#&gt; # $description: Gencode v31 GenomicState from derfinder::makeGenomicSta...</span>
<span class="co">#&gt; # $taxonomyid: 9606</span>
<span class="co">#&gt; # $genome: GRCh37</span>
<span class="co">#&gt; # $sourcetype: GTF</span>
<span class="co">#&gt; # $sourceurl: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/r...</span>
<span class="co">#&gt; # $sourcesize: NA</span>
<span class="co">#&gt; # $tags: c("Gencode", "GenomicState", "hg19", "v31") </span>
<span class="co">#&gt; # retrieve record with 'object[["AH75184"]]'</span>


## Check the metadata
<span class="kw">mcols</span>(hub_query_gs_gencode_v31_hg19)
<span class="co">#&gt; DataFrame with 1 row and 15 columns</span>
<span class="co">#&gt;                                                    title dataprovider</span>
<span class="co">#&gt;                                              &lt;character&gt;  &lt;character&gt;</span>
<span class="co">#&gt; AH75184 GenomicState for Gencode v31 on hg19 coordinates      GENCODE</span>
<span class="co">#&gt;              species taxonomyid      genome</span>
<span class="co">#&gt;          &lt;character&gt;  &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt; AH75184 Homo sapiens       9606      GRCh37</span>
<span class="co">#&gt;                                                                                                                                                                                                     description</span>
<span class="co">#&gt;                                                                                                                                                                                                     &lt;character&gt;</span>
<span class="co">#&gt; AH75184 Gencode v31 GenomicState from derfinder::makeGenomicState() on hg19 coordinates. This is useful for packages such as derfinder and derfinderPlot. For more information, check the GenomicState package.</span>
<span class="co">#&gt;         coordinate_1_based                                      maintainer</span>
<span class="co">#&gt;                  &lt;integer&gt;                                     &lt;character&gt;</span>
<span class="co">#&gt; AH75184                  1 Leonardo Collado-Torres &lt;lcolladotor@gmail.com&gt;</span>
<span class="co">#&gt;         rdatadateadded preparerclass</span>
<span class="co">#&gt;            &lt;character&gt;   &lt;character&gt;</span>
<span class="co">#&gt; AH75184     2019-10-22  GenomicState</span>
<span class="co">#&gt;                                                tags  rdataclass</span>
<span class="co">#&gt;                                              &lt;list&gt; &lt;character&gt;</span>
<span class="co">#&gt; AH75184 c("Gencode", "GenomicState", "hg19", "v31")        list</span>
<span class="co">#&gt;                                                      rdatapath</span>
<span class="co">#&gt;                                                    &lt;character&gt;</span>
<span class="co">#&gt; AH75184 GenomicState/gencode/gencode_v31_hg19_GenomicState.rda</span>
<span class="co">#&gt;                                                                                                                     sourceurl</span>
<span class="co">#&gt;                                                                                                                   &lt;character&gt;</span>
<span class="co">#&gt; AH75184 ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_31/GRCh37_mapping/gencode.v31lift37.annotation.gtf.gz</span>
<span class="co">#&gt;          sourcetype</span>
<span class="co">#&gt;         &lt;character&gt;</span>
<span class="co">#&gt; AH75184         GTF</span>

## Access the file through AnnotationHub
<span class="cf">if</span>(<span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(hub_query_gs_gencode_v31_hg19) <span class="op">==</span><span class="st"> </span><span class="dv">1</span>) {
    hub_gs_gencode_v31_hg19 &lt;-<span class="st"> </span>hub_query_gs_gencode_v31_hg19[[<span class="dv">1</span>]]

    hub_gs_gencode_v31_hg19
}
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache</span>
<span class="co">#&gt; $fullGenome</span>
<span class="co">#&gt; Loading required package: GenomicRanges</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; Loading required package: GenomeInfoDb</span>
<span class="co">#&gt; GRanges object with 659263 ranges and 5 metadata columns:</span>
<span class="co">#&gt;          seqnames            ranges strand |   theRegion         tx_id</span>
<span class="co">#&gt;             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;IntegerList&gt;</span>
<span class="co">#&gt;        1     chr1       11869-12227      + |        exon           1,2</span>
<span class="co">#&gt;        2     chr1       12228-12612      + |      intron           1,2</span>
<span class="co">#&gt;        3     chr1       12613-12721      + |        exon           1,2</span>
<span class="co">#&gt;        4     chr1       12722-12974      + |      intron           1,2</span>
<span class="co">#&gt;        5     chr1       12975-13052      + |        exon             2</span>
<span class="co">#&gt;      ...      ...               ...    ... .         ...           ...</span>
<span class="co">#&gt;   659259     chrY 59208555-59214013      * |  intergenic          &lt;NA&gt;</span>
<span class="co">#&gt;   659260     chrY 59276440-59311662      * |  intergenic          &lt;NA&gt;</span>
<span class="co">#&gt;   659261     chrY 59311997-59318040      * |  intergenic          &lt;NA&gt;</span>
<span class="co">#&gt;   659262     chrY 59318921-59330251      * |  intergenic          &lt;NA&gt;</span>
<span class="co">#&gt;   659263     chrY 59360549-59373566      * |  intergenic          &lt;NA&gt;</span>
<span class="co">#&gt;                                          tx_name          gene</span>
<span class="co">#&gt;                                  &lt;CharacterList&gt; &lt;IntegerList&gt;</span>
<span class="co">#&gt;        1 ENST00000450305.2_1,ENST00000456328.2_1         26085</span>
<span class="co">#&gt;        2 ENST00000450305.2_1,ENST00000456328.2_1         26085</span>
<span class="co">#&gt;        3 ENST00000450305.2_1,ENST00000456328.2_1         26085</span>
<span class="co">#&gt;        4 ENST00000450305.2_1,ENST00000456328.2_1         26085</span>
<span class="co">#&gt;        5                     ENST00000450305.2_1         26085</span>
<span class="co">#&gt;      ...                                     ...           ...</span>
<span class="co">#&gt;   659259                                    &lt;NA&gt;          &lt;NA&gt;</span>
<span class="co">#&gt;   659260                                    &lt;NA&gt;          &lt;NA&gt;</span>
<span class="co">#&gt;   659261                                    &lt;NA&gt;          &lt;NA&gt;</span>
<span class="co">#&gt;   659262                                    &lt;NA&gt;          &lt;NA&gt;</span>
<span class="co">#&gt;   659263                                    &lt;NA&gt;          &lt;NA&gt;</span>
<span class="co">#&gt;                   symbol</span>
<span class="co">#&gt;          &lt;CharacterList&gt;</span>
<span class="co">#&gt;        1         DDX11L1</span>
<span class="co">#&gt;        2         DDX11L1</span>
<span class="co">#&gt;        3         DDX11L1</span>
<span class="co">#&gt;        4         DDX11L1</span>
<span class="co">#&gt;        5         DDX11L1</span>
<span class="co">#&gt;      ...             ...</span>
<span class="co">#&gt;   659259            &lt;NA&gt;</span>
<span class="co">#&gt;   659260            &lt;NA&gt;</span>
<span class="co">#&gt;   659261            &lt;NA&gt;</span>
<span class="co">#&gt;   659262            &lt;NA&gt;</span>
<span class="co">#&gt;   659263            &lt;NA&gt;</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 24 sequences from hg19 genome</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $codingGenome</span>
<span class="co">#&gt; GRanges object with 878954 ranges and 5 metadata columns:</span>
<span class="co">#&gt;          seqnames            ranges strand |   theRegion         tx_id</span>
<span class="co">#&gt;             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;IntegerList&gt;</span>
<span class="co">#&gt;        1     chr1        9869-11868      + |    promoter           1,2</span>
<span class="co">#&gt;        2     chr1       11869-12227      + |        exon           1,2</span>
<span class="co">#&gt;        3     chr1       12228-12612      + |      intron           1,2</span>
<span class="co">#&gt;        4     chr1       12613-12721      + |        exon           1,2</span>
<span class="co">#&gt;        5     chr1       12722-12974      + |      intron           1,2</span>
<span class="co">#&gt;      ...      ...               ...    ... .         ...           ...</span>
<span class="co">#&gt;   878950     chrY 59208555-59212013      * |  intergenic          &lt;NA&gt;</span>
<span class="co">#&gt;   878951     chrY 59276440-59311662      * |  intergenic          &lt;NA&gt;</span>
<span class="co">#&gt;   878952     chrY 59313997-59318040      * |  intergenic          &lt;NA&gt;</span>
<span class="co">#&gt;   878953     chrY 59320921-59328251      * |  intergenic          &lt;NA&gt;</span>
<span class="co">#&gt;   878954     chrY 59362549-59373566      * |  intergenic          &lt;NA&gt;</span>
<span class="co">#&gt;                                          tx_name          gene</span>
<span class="co">#&gt;                                  &lt;CharacterList&gt; &lt;IntegerList&gt;</span>
<span class="co">#&gt;        1 ENST00000450305.2_1,ENST00000456328.2_1         26085</span>
<span class="co">#&gt;        2 ENST00000450305.2_1,ENST00000456328.2_1         26085</span>
<span class="co">#&gt;        3 ENST00000450305.2_1,ENST00000456328.2_1         26085</span>
<span class="co">#&gt;        4 ENST00000450305.2_1,ENST00000456328.2_1         26085</span>
<span class="co">#&gt;        5 ENST00000450305.2_1,ENST00000456328.2_1         26085</span>
<span class="co">#&gt;      ...                                     ...           ...</span>
<span class="co">#&gt;   878950                                    &lt;NA&gt;          &lt;NA&gt;</span>
<span class="co">#&gt;   878951                                    &lt;NA&gt;          &lt;NA&gt;</span>
<span class="co">#&gt;   878952                                    &lt;NA&gt;          &lt;NA&gt;</span>
<span class="co">#&gt;   878953                                    &lt;NA&gt;          &lt;NA&gt;</span>
<span class="co">#&gt;   878954                                    &lt;NA&gt;          &lt;NA&gt;</span>
<span class="co">#&gt;                   symbol</span>
<span class="co">#&gt;          &lt;CharacterList&gt;</span>
<span class="co">#&gt;        1         DDX11L1</span>
<span class="co">#&gt;        2         DDX11L1</span>
<span class="co">#&gt;        3         DDX11L1</span>
<span class="co">#&gt;        4         DDX11L1</span>
<span class="co">#&gt;        5         DDX11L1</span>
<span class="co">#&gt;      ...             ...</span>
<span class="co">#&gt;   878950            &lt;NA&gt;</span>
<span class="co">#&gt;   878951            &lt;NA&gt;</span>
<span class="co">#&gt;   878952            &lt;NA&gt;</span>
<span class="co">#&gt;   878953            &lt;NA&gt;</span>
<span class="co">#&gt;   878954            &lt;NA&gt;</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 24 sequences from hg19 genome</span></code></pre></div>
</div>
<div id="using-the-objects" class="section level1">
<h1 class="hasAnchor">
<a href="#using-the-objects" class="anchor"></a>Using the objects</h1>
<p>To show how we can use these objects, first we build those for Gencode version 31 on hg19 coordinates.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Load the example TxDb object
## or start from scratch with:
## txdb_v31_hg19_chr21 &lt;- gencode_txdb(version = '31', genome = 'hg19',
##     chrs = 'chr21')
txdb_v31_hg19_chr21 &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">loadDb</a></span>(
    <span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">'extdata'</span>, <span class="st">'txdb_v31_hg19_chr21.sqlite'</span>,
        <span class="dt">package =</span> <span class="st">'GenomicState'</span>)
)
<span class="co">#&gt; Loading required package: GenomicFeatures</span>
<span class="co">#&gt; Loading required package: AnnotationDbi</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'Biobase'</span>
<span class="co">#&gt; The following object is masked from 'package:AnnotationHub':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     cache</span>

## Build the GenomicState and annotated genes
genes_v31_hg19_chr21 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gencode_annotated_genes.html">gencode_annotated_genes</a></span>(txdb_v31_hg19_chr21)
<span class="co">#&gt; 2019-10-22 13:13:13 annotating the transcripts</span>
<span class="co">#&gt; No annotationPackage supplied. Trying org.Hs.eg.db.</span>
<span class="co">#&gt; Loading required package: org.Hs.eg.db</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Getting TSS and TSE.</span>
<span class="co">#&gt; Getting CSS and CSE.</span>
<span class="co">#&gt; Getting exons.</span>
<span class="co">#&gt; Annotating genes.</span>
<span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span>
gs_v31_hg19_chr21 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gencode_genomic_state.html">gencode_genomic_state</a></span>(txdb_v31_hg19_chr21)
<span class="co">#&gt; 2019-10-22 13:13:28 making the GenomicState object</span>
<span class="co">#&gt; extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; 2019-10-22 13:13:36 finding gene symbols</span>
<span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span>
<span class="co">#&gt; 2019-10-22 13:13:37 adding gene symbols to the GenomicState</span></code></pre></div>
<p>You can alternatively use the files hosted in <em><a href="https://bioconductor.org/packages/3.10/AnnotationHub">AnnotationHub</a></em> (<a href="#bib-Morgan_2019">Morgan, 2019</a>) which will be faster in general.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Create the AnnotationHub object once and re-use it to speed up things
ah &lt;-<span class="st"> </span>AnnotationHub<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">AnnotationHub</a></span>()
<span class="co">#&gt; snapshotDate(): 2019-10-22</span>

## Find the TxDb object for hg19 Gencode version 31
hub_query_txdb_gencode_v31_hg19 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/GenomicStateHub.html">GenomicStateHub</a></span>(<span class="dt">version =</span> <span class="st">'31'</span>,
    <span class="dt">genome =</span> <span class="st">'hg19'</span>,
    <span class="dt">filetype =</span> <span class="st">'TxDb'</span>, <span class="dt">ah =</span> ah)
hub_query_txdb_gencode_v31_hg19
<span class="co">#&gt; AnnotationHub with 1 record</span>
<span class="co">#&gt; # snapshotDate(): 2019-10-22 </span>
<span class="co">#&gt; # names(): AH75182</span>
<span class="co">#&gt; # $dataprovider: GENCODE</span>
<span class="co">#&gt; # $species: Homo sapiens</span>
<span class="co">#&gt; # $rdataclass: TxDb</span>
<span class="co">#&gt; # $rdatadateadded: 2019-10-22</span>
<span class="co">#&gt; # $title: TxDb for Gencode v31 on hg19 coordinates</span>
<span class="co">#&gt; # $description: Gencode v31 TxDb object on hg19 coordinates. This is us...</span>
<span class="co">#&gt; # $taxonomyid: 9606</span>
<span class="co">#&gt; # $genome: GRCh37</span>
<span class="co">#&gt; # $sourcetype: GTF</span>
<span class="co">#&gt; # $sourceurl: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/r...</span>
<span class="co">#&gt; # $sourcesize: NA</span>
<span class="co">#&gt; # $tags: c("Gencode", "GenomicState", "hg19", "v31") </span>
<span class="co">#&gt; # retrieve record with 'object[["AH75182"]]'</span>

## Now the Annotated Genes for hg19 Gencode v31
hub_query_genes_gencode_v31_hg19 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/GenomicStateHub.html">GenomicStateHub</a></span>(<span class="dt">version =</span> <span class="st">'31'</span>,
    <span class="dt">genome =</span> <span class="st">'hg19'</span>,
    <span class="dt">filetype =</span> <span class="st">'AnnotatedGenes'</span>, <span class="dt">ah =</span> ah)
hub_query_genes_gencode_v31_hg19
<span class="co">#&gt; AnnotationHub with 1 record</span>
<span class="co">#&gt; # snapshotDate(): 2019-10-22 </span>
<span class="co">#&gt; # names(): AH75183</span>
<span class="co">#&gt; # $dataprovider: GENCODE</span>
<span class="co">#&gt; # $species: Homo sapiens</span>
<span class="co">#&gt; # $rdataclass: GRanges</span>
<span class="co">#&gt; # $rdatadateadded: 2019-10-22</span>
<span class="co">#&gt; # $title: Annotated genes for Gencode v31 on hg19 coordinates</span>
<span class="co">#&gt; # $description: Gencode v31 annotated genes from bumphunter::annotateTr...</span>
<span class="co">#&gt; # $taxonomyid: 9606</span>
<span class="co">#&gt; # $genome: GRCh37</span>
<span class="co">#&gt; # $sourcetype: GTF</span>
<span class="co">#&gt; # $sourceurl: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/r...</span>
<span class="co">#&gt; # $sourcesize: NA</span>
<span class="co">#&gt; # $tags: c("Gencode", "GenomicState", "hg19", "v31") </span>
<span class="co">#&gt; # retrieve record with 'object[["AH75183"]]'</span>

## And finally the GenomicState for hg19 Gencode v31
hub_query_gs_gencode_v31_hg19 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/GenomicStateHub.html">GenomicStateHub</a></span>(<span class="dt">version =</span> <span class="st">'31'</span>,
    <span class="dt">genome =</span> <span class="st">'hg19'</span>,
    <span class="dt">filetype =</span> <span class="st">'GenomicState'</span>, <span class="dt">ah =</span> ah)
hub_query_gs_gencode_v31_hg19
<span class="co">#&gt; AnnotationHub with 1 record</span>
<span class="co">#&gt; # snapshotDate(): 2019-10-22 </span>
<span class="co">#&gt; # names(): AH75184</span>
<span class="co">#&gt; # $dataprovider: GENCODE</span>
<span class="co">#&gt; # $species: Homo sapiens</span>
<span class="co">#&gt; # $rdataclass: list</span>
<span class="co">#&gt; # $rdatadateadded: 2019-10-22</span>
<span class="co">#&gt; # $title: GenomicState for Gencode v31 on hg19 coordinates</span>
<span class="co">#&gt; # $description: Gencode v31 GenomicState from derfinder::makeGenomicSta...</span>
<span class="co">#&gt; # $taxonomyid: 9606</span>
<span class="co">#&gt; # $genome: GRCh37</span>
<span class="co">#&gt; # $sourcetype: GTF</span>
<span class="co">#&gt; # $sourceurl: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/r...</span>
<span class="co">#&gt; # $sourcesize: NA</span>
<span class="co">#&gt; # $tags: c("Gencode", "GenomicState", "hg19", "v31") </span>
<span class="co">#&gt; # retrieve record with 'object[["AH75184"]]'</span>

## If you want to access the files use the double bracket AnnotationHub syntax
## to retrieve the R objects from the web.
<span class="cf">if</span>(<span class="ot">FALSE</span>) {
    hub_txdb_gencode_v31_hg19 &lt;-<span class="st"> </span>hub_query_txdb_gencode_v31_hg19[[<span class="dv">1</span>]]
    hub_genes_gencode_v31_hg19 &lt;-<span class="st"> </span>hub_query_genes_gencode_v31_hg19[[<span class="dv">1</span>]]
    hub_gs_gencode_v31_hg19 &lt;-<span class="st"> </span>hub_query_gs_gencode_v31_hg19[[<span class="dv">1</span>]]
}</code></pre></div>
<p>Next we load a series of related packages that use the objects we created with <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> or downloaded from <em><a href="https://bioconductor.org/packages/3.10/AnnotationHub">AnnotationHub</a></em> (<a href="#bib-Morgan_2019">Morgan, 2019</a>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Load external packages
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'derfinder'</span>)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'derfinderPlot'</span>)
<span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span>
<span class="co">#&gt;   method from   </span>
<span class="co">#&gt;   +.gg   ggplot2</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'bumphunter'</span>)
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Loading required package: locfit</span>
<span class="co">#&gt; locfit 1.5-9.1    2013-03-22</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'GenomicRanges'</span>)</code></pre></div>
<p>Next we can prepare the needed for running <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">derfinderPlot::plotRegionCoverage()</a></code> where we use the <code>TxDb</code> object, the <code>GenomicState</code> and the <code>annotated genes</code> we prepared for Gencode v31 on hg19.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Some example regions from derfinder (set the chromosome lengths)
regions &lt;-<span class="st"> </span>genomeRegions<span class="op">$</span>regions[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]
<span class="kw">seqlengths</span>(regions) &lt;-<span class="st"> </span><span class="kw">seqlengths</span>(txdb_v31_hg19_chr21)[
    <span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">seqlengths</span>(regions))
]

## Annotate them
nearestAnnotation &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/bumphunter/man/matchGenes.html">matchGenes</a></span>(<span class="dt">x =</span> regions, <span class="dt">subject =</span> genes_v31_hg19_chr21)
annotatedRegions &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html">annotateRegions</a></span>(<span class="dt">regions =</span> regions, 
    <span class="dt">genomicState =</span> gs_v31_hg19_chr21<span class="op">$</span>fullGenome, <span class="dt">minoverlap =</span> <span class="dv">1</span>)
<span class="co">#&gt; 2019-10-22 13:13:51 annotateRegions: counting</span>
<span class="co">#&gt; 2019-10-22 13:13:51 annotateRegions: annotating</span>

## Obtain fullCov object
fullCov &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">'chr21'</span> =<span class="st"> </span>genomeDataRaw<span class="op">$</span>coverage)
regionCov &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html">getRegionCoverage</a></span>(<span class="dt">fullCov=</span>fullCov, <span class="dt">regions=</span>regions)
<span class="co">#&gt; 2019-10-22 13:13:52 getRegionCoverage: processing chr21</span>
<span class="co">#&gt; 2019-10-22 13:13:52 getRegionCoverage: done processing chr21</span></code></pre></div>
<p>And now we can make the example plot as shown below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## now make the plot
<span class="kw"><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage</a></span>(<span class="dt">regions=</span>regions, <span class="dt">regionCoverage=</span>regionCov, 
    <span class="dt">groupInfo=</span>genomeInfo<span class="op">$</span>pop, <span class="dt">nearestAnnotation=</span>nearestAnnotation, 
    <span class="dt">annotatedRegions=</span>annotatedRegions, <span class="dt">whichRegions=</span><span class="dv">1</span><span class="op">:</span><span class="dv">2</span>,
    <span class="dt">txdb =</span> txdb_v31_hg19_chr21)
<span class="co">#&gt; 2019-10-22 13:13:52 plotRegionCoverage: extracting Tx info</span>
<span class="co">#&gt; 2019-10-22 13:13:52 plotRegionCoverage: getting Tx plot info</span></code></pre></div>
<p><img src="GenomicState_files/figure-html/example_plot-1.png" width="700"><img src="GenomicState_files/figure-html/example_plot-2.png" width="700"><img src="GenomicState_files/figure-html/example_plot-3.png" width="700"></p>
</div>
<div id="jhpce" class="section level1">
<h1 class="hasAnchor">
<a href="#jhpce" class="anchor"></a>JHPCE</h1>
<p>You can also access the data locally using the function <code><a href="../reference/local_metadata.html">local_metadata()</a></code> which works at <a href="https://jhpce.jhu.edu/">JHPCE</a> or anywhere where you have re-created the files from this package. This returns a <code><a href="https://rdrr.io/r/base/data.frame.html">data.frame()</a></code> which you can subset. It also inclused the R code for loading the data which you can do using <code><a href="https://rdrr.io/r/base/eval.html">eval(parse(text = local_metadata()$loadCode))</a></code> as shown below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Get the local metadata
meta &lt;-<span class="st"> </span><span class="kw"><a href="../reference/local_metadata.html">local_metadata</a></span>()

## Subset to the data of interest, lets say hg19 TxDb for v31
interest &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(meta, RDataClass <span class="op">==</span><span class="st"> 'TxDb'</span> <span class="op">&amp;</span><span class="st"> </span>Tags <span class="op">==</span><span class="st"> 'Gencode:v31:hg19'</span>)

## Next you can load the data
<span class="cf">if</span>(<span class="kw"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(interest<span class="op">$</span>RDataPath)) {
    ## This only works at JHPCE
    <span class="kw"><a href="https://rdrr.io/r/base/eval.html">eval</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/parse.html">parse</a></span>(<span class="dt">text =</span> interest<span class="op">$</span>loadCode))
    
    ## Explore the loaded object (would be gencode_v31_hg19_txdb in this case)
    gencode_v31_hg19_txdb
}</code></pre></div>
</div>
<div id="build-objects" class="section level1">
<h1 class="hasAnchor">
<a href="#build-objects" class="anchor"></a>Build objects</h1>
<p>The objects provided by <code>GenomicState</code> through <em><a href="https://bioconductor.org/packages/3.10/AnnotationHub">AnnotationHub</a></em> (<a href="#bib-Morgan_2019">Morgan, 2019</a>) were built using code like the one included below which is how the Gencode version 23 for hg19 files were built.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">outdir &lt;-<span class="st"> 'gencode'</span>
<span class="kw"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span>(outdir, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>)
 
## Build and save the TxDb object
gencode_v23_hg19_txdb &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gencode_txdb.html">gencode_txdb</a></span>(<span class="st">'23'</span>, <span class="st">'hg19'</span>)
<span class="kw">saveDb</span>(gencode_v23_hg19_txdb,
    <span class="dt">file =</span> <span class="kw"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(outdir, <span class="st">'gencode_v23_hg19_txdb.sqlite'</span>))

## Build and save the annotateTranscripts output
gencode_v23_hg19_annotated_genes&lt;-<span class="st"> </span><span class="kw"><a href="../reference/gencode_annotated_genes.html">gencode_annotated_genes</a></span>(
    gencode_v23_hg19_txdb
)
<span class="kw"><a href="https://rdrr.io/r/base/save.html">save</a></span>(gencode_v23_hg19_annotated_genes,
    <span class="dt">file =</span> <span class="kw"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(outdir, <span class="st">'gencode_v23_hg19_annotated_genes.rda'</span>))

## Build and save the GenomicState
gencode_v23_hg19_GenomicState &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gencode_genomic_state.html">gencode_genomic_state</a></span>(
    gencode_v23_hg19_txdb
)
<span class="kw"><a href="https://rdrr.io/r/base/save.html">save</a></span>(gencode_v23_hg19_GenomicState,
    <span class="dt">file =</span> <span class="kw"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(outdir, <span class="st">'gencode_v23_hg19_GenomicState.rda'</span>))</code></pre></div>
<p>For more details check the source files:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## R commands for building the files:
<span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">'scripts'</span>, <span class="st">'make-data_gencode_human.R'</span>,
    <span class="dt">package =</span> <span class="st">'GenomicState'</span>)
<span class="co">#&gt; [1] "/private/var/folders/cx/n9s558kx6fb7jf5z_pgszgb80000gn/T/RtmpLnhUxz/temp_libpathda9073de6f4b/GenomicState/scripts/make-data_gencode_human.R"</span>
## The above file was created by this one:
<span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">'scripts'</span>, <span class="st">'generate_make_data_gencode_human.R'</span>,
    <span class="dt">package =</span> <span class="st">'GenomicState'</span>)
<span class="co">#&gt; [1] "/private/var/folders/cx/n9s558kx6fb7jf5z_pgszgb80000gn/T/RtmpLnhUxz/temp_libpathda9073de6f4b/GenomicState/scripts/generate_make_data_gencode_human.R"</span></code></pre></div>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.10/GenomicState">GenomicState</a></em> package <a id="cite-Collado-Torres_2019"></a>(<a href="https://github.com/LieberInstitute/GenomicState">Collado-Torres, 2019</a>) was made possible thanks to:</p>
<ul>
<li>R <a id="cite-R_Core_Team_2019"></a>(<a href="https://www.R-project.org/">R Core Team, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/BiocStyle">BiocStyle</a></em> <a id="cite-Oles_2019"></a>(<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/GenomeInfoDb">GenomeInfoDb</a></em> <a id="cite-GenomeInfoDb"></a>(<a href="https://doi.org/10.18129/B9.bioc.GenomeInfoDb">Arora, Morgan, Carlson, and Pagès, 2017</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> <a id="cite-Boettiger_2019"></a>(<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> <a id="cite-Xie_2014"></a>(<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em></li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/rtracklayer">rtracklayer</a></em> <a id="cite-Lawrence_2009"></a>(<a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract">Lawrence, Gentleman, and Carey, 2009</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo">sessioninfo</a></em> <a id="cite-Csardi_2018"></a>(<a href="https://CRAN.R-project.org/package=sessioninfo">Csárdi, core, Wickham, Chang, et al., 2018</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat">testthat</a></em> <a id="cite-Wickham_2011"></a>(<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf">Wickham, 2011</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/GenomicFeatures">GenomicFeatures</a></em> <a id="cite-Lawrence_2013"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.10/bumphunter">bumphunter</a></em></li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/derfinder">derfinder</a></em> <a id="cite-Collado-Torres_2017"></a>(<a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852">Collado-Torres, Nellore, Frazee, Wilks, et al., 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/AnnotationDbi">AnnotationDbi</a></em> <a id="cite-Pages_2019"></a>(<a href="#bib-Pages_2019">Pagès, Carlson, Falcon, and Li, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/IRanges">IRanges</a></em> <a id="cite-Lawrence_2013a"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/org.Hs.eg.db">org.Hs.eg.db</a></em> <a id="cite-Carlson_2019"></a>(<a href="#bib-Carlson_2019">Carlson, 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=glue">glue</a></em> <a id="cite-Hester_2019"></a>(<a href="https://CRAN.R-project.org/package=glue">Hester, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/AnnotationHub">AnnotationHub</a></em> (<a href="#bib-Morgan_2019">Morgan, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/AnnotationHubData">AnnotationHubData</a></em> <a id="cite-Bioconductor_Package_Maintainer_2019"></a>(<a href="#bib-Bioconductor_Package_Maintainer_2019">Bioconductor Package Maintainer, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/GenomicRanges">GenomicRanges</a></em> <a id="cite-Lawrence_2013ab"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Create the vignette
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'rmarkdown'</span>)
<span class="kw"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">'GenomicState.Rmd'</span>))

## Extract the R code
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'knitr'</span>)
<span class="kw"><a href="https://rdrr.io/pkg/knitr/man/knit.html">knit</a></span>(<span class="st">'GenomicState.Rmd'</span>, <span class="dt">tangle =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code>#&gt; [1] "2019-10-22 13:13:53 EDT"</code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code>#&gt; Time difference of 1.104 mins</code></pre>
<p><code>R</code> session information.</p>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 3.6.1 (2019-07-05)
#&gt;  os       macOS Mojave 10.14.6        
#&gt;  system   x86_64, darwin15.6.0        
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_US.UTF-8                 
#&gt;  ctype    en_US.UTF-8                 
#&gt;  tz       America/New_York            
#&gt;  date     2019-10-22                  
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package                * version   date       lib source        
#&gt;  acepack                  1.4.1     2016-10-29 [2] CRAN (R 3.6.0)
#&gt;  AnnotationDbi          * 1.47.1    2019-08-20 [2] Bioconductor  
#&gt;  AnnotationFilter         1.9.0     2019-05-02 [2] Bioconductor  
#&gt;  AnnotationHub          * 2.17.10   2019-10-01 [2] Bioconductor  
#&gt;  askpass                  1.1       2019-01-13 [2] CRAN (R 3.6.0)
#&gt;  assertthat               0.2.1     2019-03-21 [2] CRAN (R 3.6.0)
#&gt;  backports                1.1.5     2019-10-02 [2] CRAN (R 3.6.0)
#&gt;  base64enc                0.1-3     2015-07-28 [2] CRAN (R 3.6.0)
#&gt;  bibtex                   0.4.2     2017-06-30 [2] CRAN (R 3.6.0)
#&gt;  Biobase                * 2.45.1    2019-09-04 [2] Bioconductor  
#&gt;  BiocFileCache          * 1.9.1     2019-06-03 [2] Bioconductor  
#&gt;  BiocGenerics           * 0.31.6    2019-09-19 [2] Bioconductor  
#&gt;  BiocManager              1.30.8    2019-10-16 [2] CRAN (R 3.6.1)
#&gt;  BiocParallel             1.19.5    2019-10-20 [2] Bioconductor  
#&gt;  BiocStyle              * 2.13.2    2019-06-12 [2] Bioconductor  
#&gt;  biomaRt                  2.41.9    2019-10-01 [2] Bioconductor  
#&gt;  Biostrings               2.53.2    2019-07-09 [2] Bioconductor  
#&gt;  biovizBase               1.33.1    2019-08-23 [2] Bioconductor  
#&gt;  bit                      1.1-14    2018-05-29 [2] CRAN (R 3.6.0)
#&gt;  bit64                    0.9-7     2017-05-08 [2] CRAN (R 3.6.0)
#&gt;  bitops                   1.0-6     2013-08-17 [2] CRAN (R 3.6.0)
#&gt;  blob                     1.2.0     2019-07-09 [2] CRAN (R 3.6.0)
#&gt;  bookdown                 0.14      2019-10-01 [2] CRAN (R 3.6.0)
#&gt;  BSgenome                 1.53.2    2019-08-28 [2] Bioconductor  
#&gt;  bumphunter             * 1.27.0    2019-05-02 [2] Bioconductor  
#&gt;  checkmate                1.9.4     2019-07-04 [2] CRAN (R 3.6.0)
#&gt;  cli                      1.1.0     2019-03-19 [2] CRAN (R 3.6.0)
#&gt;  cluster                  2.1.0     2019-06-19 [2] CRAN (R 3.6.1)
#&gt;  codetools                0.2-16    2018-12-24 [2] CRAN (R 3.6.1)
#&gt;  colorspace               1.4-1     2019-03-18 [2] CRAN (R 3.6.0)
#&gt;  crayon                   1.3.4     2017-09-16 [2] CRAN (R 3.6.0)
#&gt;  curl                     4.2       2019-09-24 [2] CRAN (R 3.6.1)
#&gt;  data.table               1.12.6    2019-10-18 [2] CRAN (R 3.6.0)
#&gt;  DBI                      1.0.0     2018-05-02 [2] CRAN (R 3.6.0)
#&gt;  dbplyr                 * 1.4.2     2019-06-17 [2] CRAN (R 3.6.0)
#&gt;  DelayedArray             0.11.8    2019-10-08 [2] Bioconductor  
#&gt;  derfinder              * 1.19.9    2019-09-21 [2] Bioconductor  
#&gt;  derfinderHelper          1.19.3    2019-09-17 [2] Bioconductor  
#&gt;  derfinderPlot          * 1.19.3    2019-09-17 [2] Bioconductor  
#&gt;  desc                     1.2.0     2018-05-01 [2] CRAN (R 3.6.0)
#&gt;  dichromat                2.0-0     2013-01-24 [2] CRAN (R 3.6.0)
#&gt;  digest                   0.6.22    2019-10-21 [2] CRAN (R 3.6.1)
#&gt;  doRNG                    1.7.1     2018-06-22 [2] CRAN (R 3.6.0)
#&gt;  dplyr                    0.8.3     2019-07-04 [2] CRAN (R 3.6.0)
#&gt;  ensembldb                2.9.6     2019-09-10 [2] Bioconductor  
#&gt;  evaluate                 0.14      2019-05-28 [2] CRAN (R 3.6.0)
#&gt;  fastmap                  1.0.1     2019-10-08 [2] CRAN (R 3.6.0)
#&gt;  foreach                * 1.4.7     2019-07-27 [2] CRAN (R 3.6.0)
#&gt;  foreign                  0.8-72    2019-08-02 [2] CRAN (R 3.6.0)
#&gt;  Formula                  1.2-3     2018-05-03 [2] CRAN (R 3.6.0)
#&gt;  fs                       1.3.1     2019-05-06 [2] CRAN (R 3.6.0)
#&gt;  GenomeInfoDb           * 1.21.2    2019-09-27 [2] Bioconductor  
#&gt;  GenomeInfoDbData         1.2.1     2019-07-27 [2] Bioconductor  
#&gt;  GenomicAlignments        1.21.7    2019-09-11 [2] Bioconductor  
#&gt;  GenomicFeatures        * 1.37.4    2019-07-09 [2] Bioconductor  
#&gt;  GenomicFiles             1.21.0    2019-05-02 [2] Bioconductor  
#&gt;  GenomicRanges          * 1.37.17   2019-10-11 [2] Bioconductor  
#&gt;  GenomicState           * 0.99.5    2019-10-22 [1] Bioconductor  
#&gt;  GGally                   1.4.0     2018-05-17 [2] CRAN (R 3.6.0)
#&gt;  ggbio                    1.33.2    2019-08-24 [2] Bioconductor  
#&gt;  ggplot2                  3.2.1     2019-08-10 [2] CRAN (R 3.6.0)
#&gt;  glue                     1.3.1     2019-03-12 [2] CRAN (R 3.6.0)
#&gt;  graph                    1.63.0    2019-05-02 [2] Bioconductor  
#&gt;  gridExtra                2.3       2017-09-09 [2] CRAN (R 3.6.0)
#&gt;  gtable                   0.3.0     2019-03-25 [2] CRAN (R 3.6.0)
#&gt;  Hmisc                    4.2-0     2019-01-26 [2] CRAN (R 3.6.0)
#&gt;  hms                      0.5.1     2019-08-23 [2] CRAN (R 3.6.0)
#&gt;  htmlTable                1.13.2    2019-09-22 [2] CRAN (R 3.6.1)
#&gt;  htmltools                0.4.0     2019-10-04 [2] CRAN (R 3.6.0)
#&gt;  htmlwidgets              1.5.1     2019-10-08 [2] CRAN (R 3.6.0)
#&gt;  httpuv                   1.5.2     2019-09-11 [2] CRAN (R 3.6.0)
#&gt;  httr                     1.4.1     2019-08-05 [2] CRAN (R 3.6.0)
#&gt;  interactiveDisplayBase   1.23.0    2019-05-02 [2] Bioconductor  
#&gt;  IRanges                * 2.19.17   2019-10-11 [2] Bioconductor  
#&gt;  iterators              * 1.0.12    2019-07-26 [2] CRAN (R 3.6.1)
#&gt;  jsonlite                 1.6       2018-12-07 [2] CRAN (R 3.6.0)
#&gt;  knitcitations          * 1.0.10    2019-09-15 [2] CRAN (R 3.6.0)
#&gt;  knitr                    1.25      2019-09-18 [2] CRAN (R 3.6.1)
#&gt;  later                    1.0.0     2019-10-04 [2] CRAN (R 3.6.0)
#&gt;  lattice                  0.20-38   2018-11-04 [2] CRAN (R 3.6.1)
#&gt;  latticeExtra             0.6-28    2016-02-09 [2] CRAN (R 3.6.0)
#&gt;  lazyeval                 0.2.2     2019-03-15 [2] CRAN (R 3.6.0)
#&gt;  limma                    3.41.18   2019-10-13 [2] Bioconductor  
#&gt;  locfit                 * 1.5-9.1   2013-04-20 [2] CRAN (R 3.6.0)
#&gt;  lubridate                1.7.4     2018-04-11 [2] CRAN (R 3.6.0)
#&gt;  magrittr                 1.5       2014-11-22 [2] CRAN (R 3.6.0)
#&gt;  MASS                     7.3-51.4  2019-03-31 [2] CRAN (R 3.6.1)
#&gt;  Matrix                   1.2-17    2019-03-22 [2] CRAN (R 3.6.1)
#&gt;  matrixStats              0.55.0    2019-09-07 [2] CRAN (R 3.6.0)
#&gt;  memoise                  1.1.0     2017-04-21 [2] CRAN (R 3.6.0)
#&gt;  mime                     0.7       2019-06-11 [2] CRAN (R 3.6.0)
#&gt;  munsell                  0.5.0     2018-06-12 [2] CRAN (R 3.6.0)
#&gt;  nnet                     7.3-12    2016-02-02 [2] CRAN (R 3.6.1)
#&gt;  openssl                  1.4.1     2019-07-18 [2] CRAN (R 3.6.0)
#&gt;  org.Hs.eg.db           * 3.8.2     2019-07-27 [2] Bioconductor  
#&gt;  OrganismDbi              1.27.1    2019-08-17 [2] Bioconductor  
#&gt;  pillar                   1.4.2     2019-06-29 [2] CRAN (R 3.6.0)
#&gt;  pkgconfig                2.0.3     2019-09-22 [2] CRAN (R 3.6.1)
#&gt;  pkgdown                  1.4.1     2019-09-15 [2] CRAN (R 3.6.0)
#&gt;  pkgmaker                 0.27      2018-05-25 [2] CRAN (R 3.6.0)
#&gt;  plyr                     1.8.4     2016-06-08 [2] CRAN (R 3.6.0)
#&gt;  prettyunits              1.0.2     2015-07-13 [2] CRAN (R 3.6.0)
#&gt;  progress                 1.2.2     2019-05-16 [2] CRAN (R 3.6.0)
#&gt;  promises                 1.1.0     2019-10-04 [2] CRAN (R 3.6.0)
#&gt;  ProtGenerics             1.17.4    2019-08-20 [2] Bioconductor  
#&gt;  purrr                    0.3.3     2019-10-18 [2] CRAN (R 3.6.0)
#&gt;  qvalue                   2.17.0    2019-05-02 [2] Bioconductor  
#&gt;  R6                       2.4.0     2019-02-14 [2] CRAN (R 3.6.0)
#&gt;  rappdirs                 0.3.1     2016-03-28 [2] CRAN (R 3.6.0)
#&gt;  RBGL                     1.61.0    2019-05-02 [2] Bioconductor  
#&gt;  RColorBrewer             1.1-2     2014-12-07 [2] CRAN (R 3.6.0)
#&gt;  Rcpp                     1.0.2     2019-07-25 [2] CRAN (R 3.6.0)
#&gt;  RCurl                    1.95-4.12 2019-03-04 [2] CRAN (R 3.6.0)
#&gt;  RefManageR               1.2.12    2019-04-03 [2] CRAN (R 3.6.0)
#&gt;  registry                 0.5-1     2019-03-05 [2] CRAN (R 3.6.0)
#&gt;  reshape                  0.8.8     2018-10-23 [2] CRAN (R 3.6.0)
#&gt;  reshape2                 1.4.3     2017-12-11 [2] CRAN (R 3.6.0)
#&gt;  rlang                    0.4.0     2019-06-25 [2] CRAN (R 3.6.0)
#&gt;  rmarkdown                1.16      2019-10-01 [2] CRAN (R 3.6.0)
#&gt;  rngtools                 1.4       2019-07-01 [2] CRAN (R 3.6.0)
#&gt;  rpart                    4.1-15    2019-04-12 [2] CRAN (R 3.6.1)
#&gt;  rprojroot                1.3-2     2018-01-03 [2] CRAN (R 3.6.0)
#&gt;  Rsamtools                2.1.7     2019-10-09 [2] Bioconductor  
#&gt;  RSQLite                  2.1.2     2019-07-24 [2] CRAN (R 3.6.0)
#&gt;  rstudioapi               0.10      2019-03-19 [2] CRAN (R 3.6.0)
#&gt;  rtracklayer              1.45.6    2019-09-17 [2] Bioconductor  
#&gt;  S4Vectors              * 0.23.25   2019-10-02 [2] Bioconductor  
#&gt;  scales                   1.0.0     2018-08-09 [2] CRAN (R 3.6.0)
#&gt;  sessioninfo            * 1.1.1     2018-11-05 [2] CRAN (R 3.6.0)
#&gt;  shiny                    1.4.0     2019-10-10 [2] CRAN (R 3.6.0)
#&gt;  stringi                  1.4.3     2019-03-12 [2] CRAN (R 3.6.0)
#&gt;  stringr                  1.4.0     2019-02-10 [2] CRAN (R 3.6.0)
#&gt;  SummarizedExperiment     1.15.9    2019-09-11 [2] Bioconductor  
#&gt;  survival                 2.44-1.1  2019-04-01 [2] CRAN (R 3.6.1)
#&gt;  tibble                   2.1.3     2019-06-06 [2] CRAN (R 3.6.0)
#&gt;  tidyselect               0.2.5     2018-10-11 [2] CRAN (R 3.6.0)
#&gt;  VariantAnnotation        1.31.6    2019-10-11 [2] Bioconductor  
#&gt;  vctrs                    0.2.0     2019-07-05 [2] CRAN (R 3.6.0)
#&gt;  withr                    2.1.2     2018-03-15 [2] CRAN (R 3.6.0)
#&gt;  xfun                     0.10      2019-10-01 [2] CRAN (R 3.6.0)
#&gt;  XML                      3.98-1.20 2019-06-06 [2] CRAN (R 3.6.0)
#&gt;  xml2                     1.2.2     2019-08-09 [2] CRAN (R 3.6.0)
#&gt;  xtable                   1.8-4     2019-04-21 [2] CRAN (R 3.6.0)
#&gt;  XVector                  0.25.0    2019-05-02 [2] Bioconductor  
#&gt;  yaml                     2.2.0     2018-07-25 [2] CRAN (R 3.6.0)
#&gt;  zeallot                  0.1.0     2018-01-28 [2] CRAN (R 3.6.0)
#&gt;  zlibbioc                 1.31.0    2019-05-02 [2] Bioconductor  
#&gt; 
#&gt; [1] /private/var/folders/cx/n9s558kx6fb7jf5z_pgszgb80000gn/T/RtmpLnhUxz/temp_libpathda9073de6f4b
#&gt; [2] /Library/Frameworks/R.framework/Versions/3.6devel/Resources/library</code></pre>
</div>
<div id="bibliography" class="section level1">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.10/BiocStyle">BiocStyle</a></em> (<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2019</a>), <em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>) and <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> running behind the scenes.</p>
<p>Citations were made with <em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> (<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>).</p>
<p>[1] S. Arora, M. Morgan, M. Carlson, and H. Pagès. <em>GenomeInfoDb: Utilities for manipulating chromosome and other ‘seqname’ identifiers</em>. 2017. DOI: 10.18129/B9.bioc.GenomeInfoDb.</p>
<p>[2] Bioconductor Package Maintainer. <em>AnnotationHubData: Transform public data resources into Bioconductor Data Structures</em>. R package version 1.15.13. 2019.</p>
<p>[3] C. Boettiger. <em>knitcitations: Citations for ‘Knitr’ Markdown Files</em>. R package version 1.0.10. 2019. &lt;URL: <a href="https://CRAN.R-project.org/package=knitcitations" class="uri">https://CRAN.R-project.org/package=knitcitations</a>&gt;.</p>
<p>[4] M. Carlson. <em>org.Hs.eg.db: Genome wide annotation for Human</em>. R package version 3.8.2. 2019.</p>
<p>[5] L. Collado-Torres. <em>GenomicState: Build and access GenomicState objects for use with derfinder tools from sources like Gencode</em>. R package version 0.99.5. 2019. &lt;URL: <a href="https://github.com/LieberInstitute/GenomicState" class="uri">https://github.com/LieberInstitute/GenomicState</a>&gt;.</p>
<p>[6] L. Collado-Torres, A. Nellore, A. C. Frazee, C. Wilks, et al. “Flexible expressed region analysis for RNA-seq with derfinder”. In: <em>Nucl. Acids Res.</em> (2017). DOI: 10.1093/nar/gkw852. &lt;URL: <a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852" class="uri">http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852</a>&gt;.</p>
<p>[7] G. Csárdi, R. core, H. Wickham, W. Chang, et al. <em>sessioninfo: R Session Information</em>. R package version 1.1.1. 2018. &lt;URL: <a href="https://CRAN.R-project.org/package=sessioninfo" class="uri">https://CRAN.R-project.org/package=sessioninfo</a>&gt;.</p>
<p>[8] J. Hester. <em>glue: Interpreted String Literals</em>. R package version 1.3.1. 2019. &lt;URL: <a href="https://CRAN.R-project.org/package=glue" class="uri">https://CRAN.R-project.org/package=glue</a>&gt;.</p>
<p>[9] M. Lawrence, R. Gentleman, and V. Carey. “rtracklayer: an R package for interfacing with genome browsers”. In: <em>Bioinformatics</em> 25 (2009), pp. 1841-1842. DOI: 10.1093/bioinformatics/btp328. &lt;URL: <a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract" class="uri">http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract</a>&gt;.</p>
<p>[10] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[11] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[12] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[13] M. Morgan. <em>AnnotationHub: Client to access AnnotationHub resources</em>. R package version 2.17.10. 2019.</p>
<p>[14] A. Oleś, M. Morgan, and W. Huber. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.13.2. 2019. &lt;URL: <a href="https://github.com/Bioconductor/BiocStyle" class="uri">https://github.com/Bioconductor/BiocStyle</a>&gt;.</p>
<p>[15] H. Pagès, M. Carlson, S. Falcon, and N. Li. <em>AnnotationDbi: Manipulation of SQLite-based annotations in Bioconductor</em>. R package version 1.47.1. 2019.</p>
<p>[16] R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2019. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[17] H. Wickham. “testthat: Get Started with Testing”. In: <em>The R Journal</em> 3 (2011), pp. 5-10. &lt;URL: <a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="uri">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>&gt;.</p>
<p>[18] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V. Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and Hall/CRC, 2014. &lt;URL: <a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#basics">Basics</a><ul class="nav nav-pills nav-stacked">
<li><a href="#install-genomicstate">Install <em><a href="https://bioconductor-org/packages/3-10/GenomicState">GenomicState</a></em></a></li>
      <li><a href="#required-knowledge">Required knowledge</a></li>
      <li><a href="#asking-for-help">Asking for help</a></li>
      <li><a href="#citing-genomicstate">Citing <em><a href="https://bioconductor-org/packages/3-10/GenomicState">GenomicState</a></em></a></li>
      </ul>
</li>
      <li><a href="#overview">Overview</a></li>
      <li><a href="#annotationhub">AnnotationHub</a></li>
      <li><a href="#using-the-objects">Using the objects</a></li>
      <li><a href="#jhpce">JHPCE</a></li>
      <li><a href="#build-objects">Build objects</a></li>
      <li><a href="#reproducibility">Reproducibility</a></li>
      <li><a href="#bibliography">Bibliography</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
