<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to GenomicState • GenomicState</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to GenomicState">
<meta property="og:description" content="GenomicState">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicState</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.15</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/GenomicState.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/GenomicState/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to GenomicState</h1>
                        <h4 data-toc-skip class="author">Leonardo
Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
CampusCenter for Computational Biology, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">7 May 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/GenomicState/blob/HEAD/vignettes/GenomicState.Rmd" class="external-link"><code>vignettes/GenomicState.Rmd</code></a></small>
      <div class="hidden name"><code>GenomicState.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-genomicstate">Install <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em><a class="anchor" aria-label="anchor" href="#install-genomicstate"></a>
</h3>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
is a <code>R</code> package available via Bioconductor. <code>R</code>
can be installed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can install
<em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"GenomicState"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="required-knowledge">Required knowledge<a class="anchor" aria-label="anchor" href="#required-knowledge"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
is based on many other packages and in particular in those that have
implemented the infrastructure needed for dealing with annotation data.
That is, packages like <em><a href="https://bioconductor.org/packages/3.17/rtracklayer" class="external-link">rtracklayer</a></em>
that allow you to import the data. A <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
user is not expected to deal with those packages directly but will need
to be familiar with <em><a href="https://bioconductor.org/packages/3.17/derfinder" class="external-link">derfinder</a></em>
and <em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em>
to understand the results <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
generates. Furthermore, it’ll be useful for the user to know the syntax
of <em><a href="https://bioconductor.org/packages/3.17/AnnotationHub" class="external-link">AnnotationHub</a></em>
(Morgan and Shepherd, 2023) in order to query and load the data provided
by this package.</p>
<p>If you are asking yourself the question “Where do I start using
Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU" class="external-link">this
blog post</a>.</p>
</div>
<div class="section level3">
<h3 id="asking-for-help">Asking for help<a class="anchor" aria-label="anchor" href="#asking-for-help"></a>
</h3>
<p>As package developers, we try to explain clearly how to use our
packages and in which order to use the functions. But <code>R</code> and
<code>Bioconductor</code> have a steep learning curve so it is critical
to learn where to ask for help. The blog post quoted above mentions some
but we would like to highlight the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support site</a>
as the main resource for getting help regarding Bioconductor. Other
alternatives are available such as creating GitHub issues and tweeting.
However, please note that if you want to receive help you should adhere
to the <a href="http://www.bioconductor.org/help/support/posting-guide/" class="external-link">posting
guidelines</a>. It is particularly critical that you provide a small
reproducible example and your session information so package developers
can track down the source of the error.</p>
</div>
<div class="section level3">
<h3 id="citing-genomicstate">Citing <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em><a class="anchor" aria-label="anchor" href="#citing-genomicstate"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"GenomicState"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'GenomicState' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Collado-Torres L (2023). _Build and access GenomicState objects for</span></span>
<span><span class="co">#&gt;   use with derfinder tools from sources like Gencode_.</span></span>
<span><span class="co">#&gt;   doi:10.18129/B9.bioc.GenomicState</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.18129/B9.bioc.GenomicState&gt;,</span></span>
<span><span class="co">#&gt;   https://github.com/LieberInstitute/GenomicState - R package version</span></span>
<span><span class="co">#&gt;   0.99.15, &lt;http://www.bioconductor.org/packages/GenomicState&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Manual{,</span></span>
<span><span class="co">#&gt;     title = {Build and access GenomicState objects for use with derfinder tools from sources like Gencode},</span></span>
<span><span class="co">#&gt;     author = {Leonardo Collado-Torres},</span></span>
<span><span class="co">#&gt;     year = {2023},</span></span>
<span><span class="co">#&gt;     url = {http://www.bioconductor.org/packages/GenomicState},</span></span>
<span><span class="co">#&gt;     note = {https://github.com/LieberInstitute/GenomicState - R package version 0.99.15},</span></span>
<span><span class="co">#&gt;     doi = {10.18129/B9.bioc.GenomicState},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
package was developed for speeding up analyses that require these
objects and in particular those that rely on Gencode annotation data.
The package <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
provides functions for building <code>GenomicState</code> objects from
diverse annotation sources such as <a href="https://www.gencodegenes.org/human/releases.html" class="external-link"><code>Gencode</code></a>.
It also provides a way to load pre-computed <code>GenomicState</code>
objects if you are working at <a href="https://jhpce.jhu.edu/" class="external-link">JHPCE</a>. These <code>GenomicState</code>
objects are normally created using <a href="https://rdrr.io/bioc/derfinder/man/makeGenomicState.html" class="external-link">derfinder::makeGenomicState()</a>
and can be used for annotating regions with <a href="https://rdrr.io/bioc/derfinder/man/annotateRegions.html" class="external-link">derfinder::annotateRegions()</a>
which are in turn used by <a href="https://rdrr.io/bioc/derfinderPlot/man/plotRegionCoverage.html" class="external-link">derfinderPlot::plotRegionCoverage()</a>.</p>
<p>To get started, load the <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/GenomicState" class="external-link">"GenomicState"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="annotationhub">AnnotationHub<a class="anchor" aria-label="anchor" href="#annotationhub"></a>
</h2>
<p>Using the <code>GencodeStateHub()</code> function you can query and
download the data from <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
using <em><a href="https://bioconductor.org/packages/3.17/AnnotationHub" class="external-link">AnnotationHub</a></em>
(Morgan and Shepherd, 2023).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Query AnnotationHub for the GenomicState object for Gencode v31 on</span></span>
<span><span class="co">## hg19 coordinates</span></span>
<span><span class="va">hub_query_gs_gencode_v31_hg19</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GenomicStateHub.html">GenomicStateHub</a></span><span class="op">(</span></span>
<span>    version <span class="op">=</span> <span class="st">"31"</span>,</span>
<span>    genome <span class="op">=</span> <span class="st">"hg19"</span>,</span>
<span>    filetype <span class="op">=</span> <span class="st">"GenomicState"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">hub_query_gs_gencode_v31_hg19</span></span>
<span><span class="co">#&gt; AnnotationHub with 1 record</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2023-04-06</span></span>
<span><span class="co">#&gt; # names(): AH75184</span></span>
<span><span class="co">#&gt; # $dataprovider: GENCODE</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: list</span></span>
<span><span class="co">#&gt; # $rdatadateadded: 2019-10-22</span></span>
<span><span class="co">#&gt; # $title: GenomicState for Gencode v31 on hg19 coordinates</span></span>
<span><span class="co">#&gt; # $description: Gencode v31 GenomicState from derfinder::makeGenomicState() ...</span></span>
<span><span class="co">#&gt; # $taxonomyid: 9606</span></span>
<span><span class="co">#&gt; # $genome: GRCh37</span></span>
<span><span class="co">#&gt; # $sourcetype: GTF</span></span>
<span><span class="co">#&gt; # $sourceurl: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/releas...</span></span>
<span><span class="co">#&gt; # $sourcesize: NA</span></span>
<span><span class="co">#&gt; # $tags: c("Gencode", "GenomicState", "hg19", "v31") </span></span>
<span><span class="co">#&gt; # retrieve record with 'object[["AH75184"]]'</span></span>
<span></span>
<span></span>
<span><span class="co">## Check the metadata</span></span>
<span><span class="fu">mcols</span><span class="op">(</span><span class="va">hub_query_gs_gencode_v31_hg19</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 1 row and 15 columns</span></span>
<span><span class="co">#&gt;                          title dataprovider      species taxonomyid      genome</span></span>
<span><span class="co">#&gt;                    &lt;character&gt;  &lt;character&gt;  &lt;character&gt;  &lt;integer&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; AH75184 GenomicState for Gen..      GENCODE Homo sapiens       9606      GRCh37</span></span>
<span><span class="co">#&gt;                    description coordinate_1_based             maintainer</span></span>
<span><span class="co">#&gt;                    &lt;character&gt;          &lt;integer&gt;            &lt;character&gt;</span></span>
<span><span class="co">#&gt; AH75184 Gencode v31 GenomicS..                  1 Leonardo Collado-Tor..</span></span>
<span><span class="co">#&gt;         rdatadateadded preparerclass                          tags  rdataclass</span></span>
<span><span class="co">#&gt;            &lt;character&gt;   &lt;character&gt;                        &lt;AsIs&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; AH75184     2019-10-22  GenomicState Gencode,GenomicState,hg19,...        list</span></span>
<span><span class="co">#&gt;                      rdatapath              sourceurl  sourcetype</span></span>
<span><span class="co">#&gt;                    &lt;character&gt;            &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; AH75184 GenomicState/gencode.. ftp://ftp.ebi.ac.uk/..         GTF</span></span>
<span></span>
<span><span class="co">## Access the file through AnnotationHub</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">hub_query_gs_gencode_v31_hg19</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">hub_gs_gencode_v31_hg19</span> <span class="op">&lt;-</span> <span class="va">hub_query_gs_gencode_v31_hg19</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span>    <span class="va">hub_gs_gencode_v31_hg19</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; $fullGenome</span></span>
<span><span class="co">#&gt; GRanges object with 659263 ranges and 5 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames            ranges strand |   theRegion         tx_id</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;IntegerList&gt;</span></span>
<span><span class="co">#&gt;        1     chr1       11869-12227      + |        exon           1,2</span></span>
<span><span class="co">#&gt;        2     chr1       12228-12612      + |      intron           1,2</span></span>
<span><span class="co">#&gt;        3     chr1       12613-12721      + |        exon           1,2</span></span>
<span><span class="co">#&gt;        4     chr1       12722-12974      + |      intron           1,2</span></span>
<span><span class="co">#&gt;        5     chr1       12975-13052      + |        exon             2</span></span>
<span><span class="co">#&gt;      ...      ...               ...    ... .         ...           ...</span></span>
<span><span class="co">#&gt;   659259     chrY 59208555-59214013      * |  intergenic              </span></span>
<span><span class="co">#&gt;   659260     chrY 59276440-59311662      * |  intergenic              </span></span>
<span><span class="co">#&gt;   659261     chrY 59311997-59318040      * |  intergenic              </span></span>
<span><span class="co">#&gt;   659262     chrY 59318921-59330251      * |  intergenic              </span></span>
<span><span class="co">#&gt;   659263     chrY 59360549-59373566      * |  intergenic              </span></span>
<span><span class="co">#&gt;                                          tx_name          gene          symbol</span></span>
<span><span class="co">#&gt;                                  &lt;CharacterList&gt; &lt;IntegerList&gt; &lt;CharacterList&gt;</span></span>
<span><span class="co">#&gt;        1 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1</span></span>
<span><span class="co">#&gt;        2 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1</span></span>
<span><span class="co">#&gt;        3 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1</span></span>
<span><span class="co">#&gt;        4 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1</span></span>
<span><span class="co">#&gt;        5                     ENST00000450305.2_1         26085         DDX11L1</span></span>
<span><span class="co">#&gt;      ...                                     ...           ...             ...</span></span>
<span><span class="co">#&gt;   659259                                                                      </span></span>
<span><span class="co">#&gt;   659260                                                                      </span></span>
<span><span class="co">#&gt;   659261                                                                      </span></span>
<span><span class="co">#&gt;   659262                                                                      </span></span>
<span><span class="co">#&gt;   659263                                                                      </span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from hg19 genome</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $codingGenome</span></span>
<span><span class="co">#&gt; GRanges object with 878954 ranges and 5 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames            ranges strand |   theRegion         tx_id</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;IntegerList&gt;</span></span>
<span><span class="co">#&gt;        1     chr1        9869-11868      + |    promoter           1,2</span></span>
<span><span class="co">#&gt;        2     chr1       11869-12227      + |        exon           1,2</span></span>
<span><span class="co">#&gt;        3     chr1       12228-12612      + |      intron           1,2</span></span>
<span><span class="co">#&gt;        4     chr1       12613-12721      + |        exon           1,2</span></span>
<span><span class="co">#&gt;        5     chr1       12722-12974      + |      intron           1,2</span></span>
<span><span class="co">#&gt;      ...      ...               ...    ... .         ...           ...</span></span>
<span><span class="co">#&gt;   878950     chrY 59208555-59212013      * |  intergenic              </span></span>
<span><span class="co">#&gt;   878951     chrY 59276440-59311662      * |  intergenic              </span></span>
<span><span class="co">#&gt;   878952     chrY 59313997-59318040      * |  intergenic              </span></span>
<span><span class="co">#&gt;   878953     chrY 59320921-59328251      * |  intergenic              </span></span>
<span><span class="co">#&gt;   878954     chrY 59362549-59373566      * |  intergenic              </span></span>
<span><span class="co">#&gt;                                          tx_name          gene          symbol</span></span>
<span><span class="co">#&gt;                                  &lt;CharacterList&gt; &lt;IntegerList&gt; &lt;CharacterList&gt;</span></span>
<span><span class="co">#&gt;        1 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1</span></span>
<span><span class="co">#&gt;        2 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1</span></span>
<span><span class="co">#&gt;        3 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1</span></span>
<span><span class="co">#&gt;        4 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1</span></span>
<span><span class="co">#&gt;        5 ENST00000450305.2_1,ENST00000456328.2_1         26085         DDX11L1</span></span>
<span><span class="co">#&gt;      ...                                     ...           ...             ...</span></span>
<span><span class="co">#&gt;   878950                                                                      </span></span>
<span><span class="co">#&gt;   878951                                                                      </span></span>
<span><span class="co">#&gt;   878952                                                                      </span></span>
<span><span class="co">#&gt;   878953                                                                      </span></span>
<span><span class="co">#&gt;   878954                                                                      </span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from hg19 genome</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="using-the-objects">Using the objects<a class="anchor" aria-label="anchor" href="#using-the-objects"></a>
</h2>
<p>To show how we can use these objects, first we build those for
Gencode version 31 on hg19 coordinates.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load the example TxDb object</span></span>
<span><span class="co">## or start from scratch with:</span></span>
<span><span class="co">## txdb_v31_hg19_chr21 &lt;- gencode_txdb(version = '31', genome = 'hg19',</span></span>
<span><span class="co">##     chrs = 'chr21')</span></span>
<span><span class="va">txdb_v31_hg19_chr21</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">loadDb</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"txdb_v31_hg19_chr21.sqlite"</span>,</span>
<span>        package <span class="op">=</span> <span class="st">"GenomicState"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: GenomicFeatures</span></span>
<span><span class="co">#&gt; Loading required package: AnnotationDbi</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:AnnotationHub':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     cache</span></span>
<span></span>
<span><span class="co">## Build the GenomicState and annotated genes</span></span>
<span><span class="va">genes_v31_hg19_chr21</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gencode_annotated_genes.html">gencode_annotated_genes</a></span><span class="op">(</span><span class="va">txdb_v31_hg19_chr21</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2023-05-07 05:53:21.007535 annotating the transcripts</span></span>
<span><span class="co">#&gt; No annotationPackage supplied. Trying org.Hs.eg.db.</span></span>
<span><span class="co">#&gt; Loading required package: org.Hs.eg.db</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Getting TSS and TSE.</span></span>
<span><span class="co">#&gt; Getting CSS and CSE.</span></span>
<span><span class="co">#&gt; Getting exons.</span></span>
<span><span class="co">#&gt; Annotating genes.</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="va">gs_v31_hg19_chr21</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gencode_genomic_state.html">gencode_genomic_state</a></span><span class="op">(</span><span class="va">txdb_v31_hg19_chr21</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2023-05-07 05:53:41.836319 making the GenomicState object</span></span>
<span><span class="co">#&gt; extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 2023-05-07 05:53:49.164824 finding gene symbols</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; 2023-05-07 05:53:49.525492 adding gene symbols to the GenomicState</span></span></code></pre></div>
<p>You can alternatively use the files hosted in <em><a href="https://bioconductor.org/packages/3.17/AnnotationHub" class="external-link">AnnotationHub</a></em>
(Morgan and Shepherd, 2023) which will be faster in general.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the AnnotationHub object once and re-use it to speed up things</span></span>
<span><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Find the TxDb object for hg19 Gencode version 31</span></span>
<span><span class="va">hub_query_txdb_gencode_v31_hg19</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GenomicStateHub.html">GenomicStateHub</a></span><span class="op">(</span></span>
<span>    version <span class="op">=</span> <span class="st">"31"</span>,</span>
<span>    genome <span class="op">=</span> <span class="st">"hg19"</span>,</span>
<span>    filetype <span class="op">=</span> <span class="st">"TxDb"</span>, ah <span class="op">=</span> <span class="va">ah</span></span>
<span><span class="op">)</span></span>
<span><span class="va">hub_query_txdb_gencode_v31_hg19</span></span>
<span><span class="co">#&gt; AnnotationHub with 1 record</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2023-04-06</span></span>
<span><span class="co">#&gt; # names(): AH75182</span></span>
<span><span class="co">#&gt; # $dataprovider: GENCODE</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: TxDb</span></span>
<span><span class="co">#&gt; # $rdatadateadded: 2019-10-22</span></span>
<span><span class="co">#&gt; # $title: TxDb for Gencode v31 on hg19 coordinates</span></span>
<span><span class="co">#&gt; # $description: Gencode v31 TxDb object on hg19 coordinates. This is useful ...</span></span>
<span><span class="co">#&gt; # $taxonomyid: 9606</span></span>
<span><span class="co">#&gt; # $genome: GRCh37</span></span>
<span><span class="co">#&gt; # $sourcetype: GTF</span></span>
<span><span class="co">#&gt; # $sourceurl: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/releas...</span></span>
<span><span class="co">#&gt; # $sourcesize: NA</span></span>
<span><span class="co">#&gt; # $tags: c("Gencode", "GenomicState", "hg19", "v31") </span></span>
<span><span class="co">#&gt; # retrieve record with 'object[["AH75182"]]'</span></span>
<span></span>
<span><span class="co">## Now the Annotated Genes for hg19 Gencode v31</span></span>
<span><span class="va">hub_query_genes_gencode_v31_hg19</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GenomicStateHub.html">GenomicStateHub</a></span><span class="op">(</span></span>
<span>    version <span class="op">=</span> <span class="st">"31"</span>,</span>
<span>    genome <span class="op">=</span> <span class="st">"hg19"</span>,</span>
<span>    filetype <span class="op">=</span> <span class="st">"AnnotatedGenes"</span>, ah <span class="op">=</span> <span class="va">ah</span></span>
<span><span class="op">)</span></span>
<span><span class="va">hub_query_genes_gencode_v31_hg19</span></span>
<span><span class="co">#&gt; AnnotationHub with 1 record</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2023-04-06</span></span>
<span><span class="co">#&gt; # names(): AH75183</span></span>
<span><span class="co">#&gt; # $dataprovider: GENCODE</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges</span></span>
<span><span class="co">#&gt; # $rdatadateadded: 2019-10-22</span></span>
<span><span class="co">#&gt; # $title: Annotated genes for Gencode v31 on hg19 coordinates</span></span>
<span><span class="co">#&gt; # $description: Gencode v31 annotated genes from bumphunter::annotateTranscr...</span></span>
<span><span class="co">#&gt; # $taxonomyid: 9606</span></span>
<span><span class="co">#&gt; # $genome: GRCh37</span></span>
<span><span class="co">#&gt; # $sourcetype: GTF</span></span>
<span><span class="co">#&gt; # $sourceurl: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/releas...</span></span>
<span><span class="co">#&gt; # $sourcesize: NA</span></span>
<span><span class="co">#&gt; # $tags: c("Gencode", "GenomicState", "hg19", "v31") </span></span>
<span><span class="co">#&gt; # retrieve record with 'object[["AH75183"]]'</span></span>
<span></span>
<span><span class="co">## And finally the GenomicState for hg19 Gencode v31</span></span>
<span><span class="va">hub_query_gs_gencode_v31_hg19</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GenomicStateHub.html">GenomicStateHub</a></span><span class="op">(</span></span>
<span>    version <span class="op">=</span> <span class="st">"31"</span>,</span>
<span>    genome <span class="op">=</span> <span class="st">"hg19"</span>,</span>
<span>    filetype <span class="op">=</span> <span class="st">"GenomicState"</span>, ah <span class="op">=</span> <span class="va">ah</span></span>
<span><span class="op">)</span></span>
<span><span class="va">hub_query_gs_gencode_v31_hg19</span></span>
<span><span class="co">#&gt; AnnotationHub with 1 record</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2023-04-06</span></span>
<span><span class="co">#&gt; # names(): AH75184</span></span>
<span><span class="co">#&gt; # $dataprovider: GENCODE</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: list</span></span>
<span><span class="co">#&gt; # $rdatadateadded: 2019-10-22</span></span>
<span><span class="co">#&gt; # $title: GenomicState for Gencode v31 on hg19 coordinates</span></span>
<span><span class="co">#&gt; # $description: Gencode v31 GenomicState from derfinder::makeGenomicState() ...</span></span>
<span><span class="co">#&gt; # $taxonomyid: 9606</span></span>
<span><span class="co">#&gt; # $genome: GRCh37</span></span>
<span><span class="co">#&gt; # $sourcetype: GTF</span></span>
<span><span class="co">#&gt; # $sourceurl: ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/releas...</span></span>
<span><span class="co">#&gt; # $sourcesize: NA</span></span>
<span><span class="co">#&gt; # $tags: c("Gencode", "GenomicState", "hg19", "v31") </span></span>
<span><span class="co">#&gt; # retrieve record with 'object[["AH75184"]]'</span></span>
<span></span>
<span><span class="co">## If you want to access the files use the double bracket AnnotationHub syntax</span></span>
<span><span class="co">## to retrieve the R objects from the web.</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">hub_txdb_gencode_v31_hg19</span> <span class="op">&lt;-</span> <span class="va">hub_query_txdb_gencode_v31_hg19</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">hub_genes_gencode_v31_hg19</span> <span class="op">&lt;-</span> <span class="va">hub_query_genes_gencode_v31_hg19</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">hub_gs_gencode_v31_hg19</span> <span class="op">&lt;-</span> <span class="va">hub_query_gs_gencode_v31_hg19</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Next we load a series of related packages that use the objects we
created with <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
or downloaded from <em><a href="https://bioconductor.org/packages/3.17/AnnotationHub" class="external-link">AnnotationHub</a></em>
(Morgan and Shepherd, 2023).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load external packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lcolladotor/derfinder" class="external-link">"derfinder"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/derfinderPlot" class="external-link">"derfinderPlot"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span></span>
<span><span class="co">#&gt;   method from   </span></span>
<span><span class="co">#&gt;   +.gg   ggplot2</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rafalab/bumphunter" class="external-link">"bumphunter"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: foreach</span></span>
<span><span class="co">#&gt; Loading required package: iterators</span></span>
<span><span class="co">#&gt; Loading required package: parallel</span></span>
<span><span class="co">#&gt; Loading required package: locfit</span></span>
<span><span class="co">#&gt; locfit 1.5-9.7    2023-01-02</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">"GenomicRanges"</a></span><span class="op">)</span></span></code></pre></div>
<p>Next we can prepare the needed for running
<code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html" class="external-link">derfinderPlot::plotRegionCoverage()</a></code> where we use the
<code>TxDb</code> object, the <code>GenomicState</code> and the
<code>annotated genes</code> we prepared for Gencode v31 on hg19.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Some example regions from derfinder (set the chromosome lengths)</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="va">genomeRegions</span><span class="op">$</span><span class="va">regions</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">txdb_v31_hg19_chr21</span><span class="op">)</span><span class="op">[</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Annotate them</span></span>
<span><span class="va">nearestAnnotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bumphunter/man/matchGenes.html" class="external-link">matchGenes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">regions</span>, subject <span class="op">=</span> <span class="va">genes_v31_hg19_chr21</span><span class="op">)</span></span>
<span><span class="va">annotatedRegions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html" class="external-link">annotateRegions</a></span><span class="op">(</span></span>
<span>    regions <span class="op">=</span> <span class="va">regions</span>,</span>
<span>    genomicState <span class="op">=</span> <span class="va">gs_v31_hg19_chr21</span><span class="op">$</span><span class="va">fullGenome</span>, minoverlap <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 2023-05-07 05:54:01.987304 annotateRegions: counting</span></span>
<span><span class="co">#&gt; 2023-05-07 05:54:02.064518 annotateRegions: annotating</span></span>
<span></span>
<span><span class="co">## Obtain fullCov object</span></span>
<span><span class="va">fullCov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"chr21"</span> <span class="op">=</span> <span class="va">genomeDataRaw</span><span class="op">$</span><span class="va">coverage</span><span class="op">)</span></span>
<span><span class="va">regionCov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html" class="external-link">getRegionCoverage</a></span><span class="op">(</span>fullCov <span class="op">=</span> <span class="va">fullCov</span>, regions <span class="op">=</span> <span class="va">regions</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2023-05-07 05:54:02.189553 getRegionCoverage: processing chr21</span></span>
<span><span class="co">#&gt; 2023-05-07 05:54:02.239155 getRegionCoverage: done processing chr21</span></span></code></pre></div>
<p>And now we can make the example plot as shown below.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## now make the plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html" class="external-link">plotRegionCoverage</a></span><span class="op">(</span></span>
<span>    regions <span class="op">=</span> <span class="va">regions</span>, regionCoverage <span class="op">=</span> <span class="va">regionCov</span>,</span>
<span>    groupInfo <span class="op">=</span> <span class="va">genomeInfo</span><span class="op">$</span><span class="va">pop</span>, nearestAnnotation <span class="op">=</span> <span class="va">nearestAnnotation</span>,</span>
<span>    annotatedRegions <span class="op">=</span> <span class="va">annotatedRegions</span>, whichRegions <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,</span>
<span>    txdb <span class="op">=</span> <span class="va">txdb_v31_hg19_chr21</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicState_files/figure-html/example_plot-1.png" width="700"><img src="GenomicState_files/figure-html/example_plot-2.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="jhpce">JHPCE<a class="anchor" aria-label="anchor" href="#jhpce"></a>
</h2>
<p>You can also access the data locally using the function
<code><a href="../reference/local_metadata.html">local_metadata()</a></code> which works at <a href="https://jhpce.jhu.edu/" class="external-link">JHPCE</a> or anywhere where you have
re-created the files from this package. This returns a
<code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> which you can subset. It also inclused the R
code for loading the data which you can do using
<code>eval(parse(text = local_metadata()$loadCode))</code> as shown
below.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get the local metadata</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/local_metadata.html">local_metadata</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Subset to the data of interest, lets say hg19 TxDb for v31</span></span>
<span><span class="va">interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">meta</span>, <span class="va">RDataClass</span> <span class="op">==</span> <span class="st">"TxDb"</span> <span class="op">&amp;</span> <span class="va">Tags</span> <span class="op">==</span> <span class="st">"Gencode:v31:hg19"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Next you can load the data</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">interest</span><span class="op">$</span><span class="va">RDataPath</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co">## This only works at JHPCE</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/eval.html" class="external-link">eval</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/parse.html" class="external-link">parse</a></span><span class="op">(</span>text <span class="op">=</span> <span class="va">interest</span><span class="op">$</span><span class="va">loadCode</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co">## Explore the loaded object (would be gencode_v31_hg19_txdb in this case)</span></span>
<span>    <span class="va">gencode_v31_hg19_txdb</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="build-objects">Build objects<a class="anchor" aria-label="anchor" href="#build-objects"></a>
</h2>
<p>The objects provided by <code>GenomicState</code> through <em><a href="https://bioconductor.org/packages/3.17/AnnotationHub" class="external-link">AnnotationHub</a></em>
(Morgan and Shepherd, 2023) were built using code like the one included
below which is how the Gencode version 23 for hg19 files were built.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outdir</span> <span class="op">&lt;-</span> <span class="st">"gencode"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">outdir</span>, showWarnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Build and save the TxDb object</span></span>
<span><span class="va">gencode_v23_hg19_txdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gencode_txdb.html">gencode_txdb</a></span><span class="op">(</span><span class="st">"23"</span>, <span class="st">"hg19"</span><span class="op">)</span></span>
<span><span class="fu">saveDb</span><span class="op">(</span><span class="va">gencode_v23_hg19_txdb</span>,</span>
<span>    file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outdir</span>, <span class="st">"gencode_v23_hg19_txdb.sqlite"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Build and save the annotateTranscripts output</span></span>
<span><span class="va">gencode_v23_hg19_annotated_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gencode_annotated_genes.html">gencode_annotated_genes</a></span><span class="op">(</span></span>
<span>    <span class="va">gencode_v23_hg19_txdb</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">gencode_v23_hg19_annotated_genes</span>,</span>
<span>    file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outdir</span>, <span class="st">"gencode_v23_hg19_annotated_genes.rda"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Build and save the GenomicState</span></span>
<span><span class="va">gencode_v23_hg19_GenomicState</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gencode_genomic_state.html">gencode_genomic_state</a></span><span class="op">(</span></span>
<span>    <span class="va">gencode_v23_hg19_txdb</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">gencode_v23_hg19_GenomicState</span>,</span>
<span>    file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outdir</span>, <span class="st">"gencode_v23_hg19_GenomicState.rda"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>For more details check the source files:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## R commands for building the files:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"scripts"</span>, <span class="st">"make-data_gencode_human.R"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"GenomicState"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/__w/_temp/Library/GenomicState/scripts/make-data_gencode_human.R"</span></span>
<span><span class="co">## The above file was created by this one:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"scripts"</span>, <span class="st">"generate_make_data_gencode_human.R"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"GenomicState"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/__w/_temp/Library/GenomicState/scripts/generate_make_data_gencode_human.R"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.17/GenomicState" class="external-link">GenomicState</a></em>
package (Collado-Torres, 2023) was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/GenomeInfoDb" class="external-link">GenomeInfoDb</a></em>
(Arora, Morgan, Carlson, and Pagès, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2014)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/rtracklayer" class="external-link">rtracklayer</a></em>
(Lawrence, Gentleman, and Carey, 2009)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/GenomicFeatures" class="external-link">GenomicFeatures</a></em>
(Lawrence, Huber, Pagès, Aboyoun, Carlson, Gentleman, Morgan, and Carey,
2013)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/bumphunter" class="external-link">bumphunter</a></em>
(Jaffe, Murakami, Lee, Leek, Fallin, Feinberg, and Irizarry, 2012)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/derfinder" class="external-link">derfinder</a></em>
(Collado-Torres, Nellore, Frazee, Wilks, Love, Langmead, Irizarry, Leek,
and Jaffe, 2017)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/AnnotationDbi" class="external-link">AnnotationDbi</a></em>
(Pagès, Carlson, Falcon, and Li, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/IRanges" class="external-link">IRanges</a></em>
(Lawrence, Huber, Pagès et al., 2013)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/org.Hs.eg.db" class="external-link">org.Hs.eg.db</a></em>
(Carlson, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=glue" class="external-link">glue</a></em>
(Hester and Bryan, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/AnnotationHub" class="external-link">AnnotationHub</a></em>
(Morgan and Shepherd, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/AnnotationHubData" class="external-link">AnnotationHubData</a></em>
(Bioconductor Package Maintainer, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/GenomicRanges" class="external-link">GenomicRanges</a></em>
(Lawrence, Huber, Pagès et al., 2013)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"GenomicState.Rmd"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"GenomicState.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2023-05-07 05:54:03 UTC"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 1.051 mins</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2023-05-07</span></span>
<span><span class="co">#&gt;  pandoc   2.19.2 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package               </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi          * 1.62.1    <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  AnnotationFilter         1.24.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  AnnotationHub          * 3.8.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  backports                1.4.1     <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  base64enc                0.1-3     <span style="color: #555555;">2015-07-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bibtex                   0.5.1     <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  Biobase                * 2.60.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocFileCache          * 2.8.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocGenerics           * 0.46.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocIO                   1.10.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocManager              1.30.20   <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  BiocParallel             1.34.1    <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocStyle              * 2.28.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocVersion              3.17.1    <span style="color: #555555;">2022-11-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  biomaRt                  2.56.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings               2.68.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  biovizBase               1.48.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bit                      4.0.5     <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bit64                    4.0.5     <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bitops                   1.0-7     <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  blob                     1.2.4     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bookdown                 0.33      <span style="color: #555555;">2023-03-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  BSgenome                 1.68.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bslib                    0.4.2     <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bumphunter             * 1.42.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  cachem                   1.0.8     <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  checkmate                2.2.0     <span style="color: #555555;">2023-04-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  cli                      3.6.1     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  cluster                  2.1.4     <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  codetools                0.2-19    <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  colorspace               2.1-0     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  crayon                   1.5.2     <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  curl                     5.0.0     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  data.table               1.14.8    <span style="color: #555555;">2023-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  DBI                      1.1.3     <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  dbplyr                 * 2.3.2     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  DelayedArray             0.26.2    <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  derfinder              * 1.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  derfinderHelper          1.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  derfinderPlot          * 1.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  desc                     1.4.2     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  dichromat                2.0-0.1   <span style="color: #555555;">2022-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  digest                   0.6.31    <span style="color: #555555;">2022-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  doRNG                    1.8.6     <span style="color: #555555;">2023-01-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  dplyr                    1.1.2     <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  ellipsis                 0.3.2     <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  ensembldb                2.24.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  evaluate                 0.20      <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  fansi                    1.0.4     <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  fastmap                  1.1.1     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  filelock                 1.0.2     <span style="color: #555555;">2018-10-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  foreach                * 1.5.2     <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  foreign                  0.8-84    <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  Formula                  1.2-5     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  fs                       1.6.2     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  generics                 0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb           * 1.36.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData         1.2.10    <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicAlignments        1.36.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicFeatures        * 1.52.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicFiles             1.36.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicRanges          * 1.52.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicState           * 0.99.15   <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GGally                   2.1.2     <span style="color: #555555;">2021-06-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  ggbio                    1.48.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ggplot2                  3.4.2     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  glue                     1.6.2     <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  graph                    1.78.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  gridExtra                2.3       <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  gtable                   0.3.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  highr                    0.10      <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  Hmisc                    5.0-1     <span style="color: #555555;">2023-03-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  hms                      1.1.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  htmlTable                2.4.1     <span style="color: #555555;">2022-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  htmltools                0.5.5     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets              1.6.2     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  httpuv                   1.6.9     <span style="color: #555555;">2023-02-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  httr                     1.4.5     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  interactiveDisplayBase   1.38.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  IRanges                * 2.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  iterators              * 1.0.14    <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  jquerylib                0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite                 1.8.4     <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  KEGGREST                 1.40.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  knitr                    1.42      <span style="color: #555555;">2023-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  later                    1.3.1     <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  lattice                  0.21-8    <span style="color: #555555;">2023-04-05</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  lazyeval                 0.2.2     <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle                1.0.3     <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  limma                    3.56.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  locfit                 * 1.5-9.7   <span style="color: #555555;">2023-01-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  lubridate                1.9.2     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  magrittr                 2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  Matrix                   1.5-4     <span style="color: #555555;">2023-04-04</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics           1.12.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  matrixStats              0.63.0    <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  memoise                  2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  mime                     0.12      <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  munsell                  0.5.0     <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  nnet                     7.3-19    <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  org.Hs.eg.db           * 3.17.0    <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  OrganismDbi              1.42.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  pillar                   1.9.0     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig                2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                  2.0.7     <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  plyr                     1.8.8     <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  png                      0.1-8     <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  prettyunits              1.1.1     <span style="color: #555555;">2020-01-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  progress                 1.2.2     <span style="color: #555555;">2019-05-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  promises                 1.2.0.1   <span style="color: #555555;">2021-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  ProtGenerics             1.32.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  purrr                    1.0.1     <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  qvalue                   2.32.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  R6                       2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  ragg                     1.2.5     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rappdirs                 0.3.3     <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  RBGL                     1.76.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  RColorBrewer             1.1-3     <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp                     1.0.10    <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  RCurl                    1.98-1.12 <span style="color: #555555;">2023-03-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  RefManageR             * 1.4.0     <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  reshape                  0.8.9     <span style="color: #555555;">2022-04-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  reshape2                 1.4.4     <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  restfulr                 0.0.15    <span style="color: #555555;">2022-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rjson                    0.2.21    <span style="color: #555555;">2022-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rlang                    1.1.1     <span style="color: #555555;">2023-04-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown                2.21      <span style="color: #555555;">2023-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rngtools                 1.5.2     <span style="color: #555555;">2021-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rpart                    4.1.19    <span style="color: #555555;">2022-10-21</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rprojroot                2.0.3     <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  Rsamtools                2.16.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  RSQLite                  2.3.1     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rstudioapi               0.14      <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rtracklayer              1.60.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  S4Arrays                 1.0.1     <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  S4Vectors              * 0.38.1    <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sass                     0.4.6     <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  scales                   1.2.1     <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo            * 1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  shiny                    1.7.4     <span style="color: #555555;">2022-12-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  stringi                  1.7.12    <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  stringr                  1.5.0     <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment     1.30.1    <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  systemfonts              1.0.4     <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  textshaping              0.3.6     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  tibble                   3.2.1     <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect               1.2.0     <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  timechange               0.2.0     <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  utf8                     1.2.3     <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  VariantAnnotation        1.46.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  vctrs                    0.6.2     <span style="color: #555555;">2023-04-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  withr                    2.5.0     <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  xfun                     0.39      <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  XML                      3.99-0.14 <span style="color: #555555;">2023-03-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  xml2                     1.3.4     <span style="color: #555555;">2023-04-27</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  xtable                   1.8-4     <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  XVector                  0.40.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml                     2.3.7     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc                 1.46.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.17/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2023), <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2014) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2023) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2023rmarkdown"></a><a href="#cite-allaire2023rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.21. 2023. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-GenomeInfoDb"></a><a href="#cite-GenomeInfoDb">[2]</a><cite>
S. Arora, M. Morgan, M. Carlson, et al. <em>GenomeInfoDb: Utilities for
manipulating chromosome and other ‘seqname’ identifiers</em>. 2017. DOI:
<a href="https://doi.org/10.18129/B9.bioc.GenomeInfoDb" class="external-link">10.18129/B9.bioc.GenomeInfoDb</a>.</cite>
</p>
<p>
<a id="bib-2023annotationhubdata"></a><a href="#cite-2023annotationhubdata">[3]</a><cite>
Bioconductor Package Maintainer. <em>AnnotationHubData: Transform public
data resources into Bioconductor Data Structures</em>. R package version
1.30.0. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.AnnotationHubData" class="external-link">10.18129/B9.bioc.AnnotationHubData</a>.
URL:
<a href="https://bioconductor.org/packages/AnnotationHubData" class="external-link">https://bioconductor.org/packages/AnnotationHubData</a>.</cite>
</p>
<p>
<a id="bib-carlson2023genome"></a><a href="#cite-carlson2023genome">[4]</a><cite>
M. Carlson. <em>org.Hs.eg.db: Genome wide annotation for Human</em>. R
package version 3.17.0. 2023.</cite>
</p>
<p>
<a id="bib-colladotorres2023build"></a><a href="#cite-colladotorres2023build">[5]</a><cite>
L. Collado-Torres. <em>Build and access GenomicState objects for use
with derfinder tools from sources like Gencode</em>. <a href="https://github.com/LieberInstitute/GenomicState" class="external-link uri">https://github.com/LieberInstitute/GenomicState</a> - R
package version 0.99.15. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.GenomicState" class="external-link">10.18129/B9.bioc.GenomicState</a>.
URL:
<a href="http://www.bioconductor.org/packages/GenomicState" class="external-link">http://www.bioconductor.org/packages/GenomicState</a>.</cite>
</p>
<p>
<a id="bib-colladotorres2017flexible"></a><a href="#cite-colladotorres2017flexible">[6]</a><cite>
L. Collado-Torres, A. Nellore, A. C. Frazee, et al. “Flexible expressed
region analysis for RNA-seq with derfinder”. In: <em>Nucl. Acids
Res.</em> (2017). DOI:
<a href="https://doi.org/10.1093/nar/gkw852" class="external-link">10.1093/nar/gkw852</a>.
URL:
<a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852" class="external-link">http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852</a>.</cite>
</p>
<p>
<a id="bib-hester2022glue"></a><a href="#cite-hester2022glue">[7]</a><cite>
J. Hester and J. Bryan. <em>glue: Interpreted String Literals</em>. <a href="https://github.com/tidyverse/glue" class="external-link uri">https://github.com/tidyverse/glue</a>, <a href="https://glue.tidyverse.org/" class="external-link uri">https://glue.tidyverse.org/</a>. 2022.</cite>
</p>
<p>
<a id="bib-bumphunter"></a><a href="#cite-bumphunter">[8]</a><cite> A.
E. Jaffe, P. Murakami, H. Lee, et al. “Bump hunting to identify
differentially methylated regions in epigenetic epidemiology studies”.
In: <em>International journal of epidemiology</em> 41.1 (2012),
pp. 200–209. DOI:
<a href="https://doi.org/10.1093/ije/dyr238" class="external-link">10.1093/ije/dyr238</a>.</cite>
</p>
<p>
<a id="bib-lawrence2009rtracklayer"></a><a href="#cite-lawrence2009rtracklayer">[9]</a><cite>
M. Lawrence, R. Gentleman, and V. Carey. “rtracklayer: an R package for
interfacing with genome browsers”. In: <em>Bioinformatics</em> 25
(2009), pp. 1841-1842. DOI:
<a href="https://doi.org/10.1093/bioinformatics/btp328" class="external-link">10.1093/bioinformatics/btp328</a>.
URL:
<a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract" class="external-link">http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract</a>.</cite>
</p>
<p>
<a id="bib-lawrence2013software"></a><a href="#cite-lawrence2013software">[10]</a><cite>
M. Lawrence, W. Huber, H. Pagès, et al. “Software for Computing and
Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8
2013). DOI:
<a href="https://doi.org/10.1371/journal.pcbi.1003118" class="external-link">10.1371/journal.pcbi.1003118</a>.
URL:
<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118%7D." class="external-link"></a><a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="external-link uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.
</cite></p>


<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[11]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-morgan2023annotationhub"></a><a href="#cite-morgan2023annotationhub">[12]</a><cite>
M. Morgan and L. Shepherd. <em>AnnotationHub: Client to access
AnnotationHub resources</em>. R package version 3.8.0. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.AnnotationHub" class="external-link">10.18129/B9.bioc.AnnotationHub</a>.
URL:
<a href="https://bioconductor.org/packages/AnnotationHub" class="external-link">https://bioconductor.org/packages/AnnotationHub</a>.</cite>
</p>
<p>
<a id="bib-ole2023biocstyle"></a><a href="#cite-ole2023biocstyle">[13]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.28.0. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-pags2023annotationdbi"></a><a href="#cite-pags2023annotationdbi">[14]</a><cite>
H. Pagès, M. Carlson, S. Falcon, et al. <em>AnnotationDbi: Manipulation
of SQLite-based annotations in Bioconductor</em>. R package version
1.62.1. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.AnnotationDbi" class="external-link">10.18129/B9.bioc.AnnotationDbi</a>.
URL:
<a href="https://bioconductor.org/packages/AnnotationDbi" class="external-link">https://bioconductor.org/packages/AnnotationDbi</a>.</cite>
</p>
<p>
<a id="bib-2023language"></a><a href="#cite-2023language">[15]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2023. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[16]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[17]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. <a href="https://github.com/r-lib/sessioninfo#readme" class="external-link uri">https://github.com/r-lib/sessioninfo#readme</a>, <a href="https://r-lib.github.io/sessioninfo/" class="external-link uri">https://r-lib.github.io/sessioninfo/</a>. 2021.</cite>
</p>
<p>
<a id="bib-xie2014knitr"></a><a href="#cite-xie2014knitr">[18]</a><cite>
Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”.
In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
